sql/tests: TestErrorDuringExtendedProtocolCommit failed

sql/tests.TestErrorDuringExtendedProtocolCommit [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4215034&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4215034&tab=artifacts#/) on master @ [5216fba6daf341de084498b544d4b89e97e54d6c](https://github.com/cockroachdb/cockroach/commits/5216fba6daf341de084498b544d4b89e97e54d6c):


```
=== RUN   TestErrorDuringExtendedProtocolCommit
    autocommit_extended_protocol_test.go:117: 
        	Error Trace:	autocommit_extended_protocol_test.go:117
        	Error:      	An error is expected but got nil.
        	Test:       	TestErrorDuringExtendedProtocolCommit
--- FAIL: TestErrorDuringExtendedProtocolCommit (1.62s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestErrorDuringExtendedProtocolCommit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestErrorDuringExtendedProtocolCommit [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4759094&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4759094&tab=artifacts#/) on master @ [fa0291175ecb983301d980247ad2ec6c7b233660](https://github.com/cockroachdb/cockroach/commits/fa0291175ecb983301d980247ad2ec6c7b233660):


```
=== RUN   TestErrorDuringExtendedProtocolCommit
    autocommit_extended_protocol_test.go:117: 
        	Error Trace:	autocommit_extended_protocol_test.go:117
        	Error:      	An error is expected but got nil.
        	Test:       	TestErrorDuringExtendedProtocolCommit
--- FAIL: TestErrorDuringExtendedProtocolCommit (1.62s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestErrorDuringExtendedProtocolCommit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestErrorDuringExtendedProtocolCommit [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7614283?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7614283?buildTab=artifacts#/) on master @ [cfb5ae9a96e1770daa4aef1615a46e212b561a84](https://github.com/cockroachdb/cockroach/commits/cfb5ae9a96e1770daa4aef1615a46e212b561a84):


```
W221120 13:29:28.039992 1837 kv/kvserver/spanset/batch.go:461  [-] 1122  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.040105 1837 kv/kvserver/spanset/batch.go:461  [-] 1123  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.040214 1837 kv/kvserver/spanset/batch.go:461  [-] 1124  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.040323 1837 kv/kvserver/spanset/batch.go:461  [-] 1125  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.040438 1837 kv/kvserver/spanset/batch.go:461  [-] 1126  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.040552 1837 kv/kvserver/spanset/batch.go:461  [-] 1127  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.040681 1837 kv/kvserver/spanset/batch.go:461  [-] 1128  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.040818 1837 kv/kvserver/spanset/batch.go:461  [-] 1129  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.040917 1837 kv/kvserver/spanset/batch.go:461  [-] 1130  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.041048 1837 kv/kvserver/spanset/batch.go:461  [-] 1131  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.041198 1837 kv/kvserver/spanset/batch.go:461  [-] 1132  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.041344 1837 kv/kvserver/spanset/batch.go:461  [-] 1133  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.041484 1837 kv/kvserver/spanset/batch.go:461  [-] 1134  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.041578 1837 kv/kvserver/spanset/batch.go:461  [-] 1135  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.041694 1837 kv/kvserver/spanset/batch.go:461  [-] 1136  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.041822 1837 kv/kvserver/spanset/batch.go:461  [-] 1137  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.042021 1837 kv/kvserver/spanset/batch.go:461  [-] 1138  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.042159 1837 kv/kvserver/spanset/batch.go:461  [-] 1139  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.042314 1837 kv/kvserver/spanset/batch.go:461  [-] 1140  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.042452 1837 kv/kvserver/spanset/batch.go:461  [-] 1141  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.042595 1837 kv/kvserver/spanset/batch.go:461  [-] 1142  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.042728 1837 kv/kvserver/spanset/batch.go:461  [-] 1143  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.042914 1837 kv/kvserver/spanset/batch.go:461  [-] 1144  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.043057 1837 kv/kvserver/spanset/batch.go:461  [-] 1145  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.043187 1837 kv/kvserver/spanset/batch.go:461  [-] 1146  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.043327 1837 kv/kvserver/spanset/batch.go:461  [-] 1147  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.043480 1837 kv/kvserver/spanset/batch.go:461  [-] 1148  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.043613 1837 kv/kvserver/spanset/batch.go:461  [-] 1149  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.043730 1837 kv/kvserver/spanset/batch.go:461  [-] 1150  cannot do strict timestamp checking of EngineIterator, resorting to best effort
I221120 13:29:28.112430 1163 3@pebble/event.go:677  [n1,s1,pebble] 1151  [JOB 8] WAL created 000005
I221120 13:29:28.158898 980 1@server/server_sql.go:1535  [n1] 1152  serving sql connections
I221120 13:29:28.164441 980 testutils/testcluster/testcluster.go:1360  [-] 1153  WaitForFullReplication
I221120 13:29:28.164690 980 testutils/testcluster/testcluster.go:1364  [-] 1154  WaitForFullReplication took: 1.068µs
W221120 13:29:28.166581 1907 kv/kvserver/spanset/batch.go:461  [-] 1155  cannot do strict timestamp checking of EngineIterator, resorting to best effort
W221120 13:29:28.171151 1907 kv/kvserver/spanset/batch.go:461  [-] 1156  cannot do strict timestamp checking of EngineIterator, resorting to best effort
    autocommit_extended_protocol_test.go:117: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2623/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/tests/autocommit_extended_protocol_test.go:117
        	Error:      	An error is expected but got nil.
        	Test:       	TestErrorDuringExtendedProtocolCommit
E221120 13:29:28.199544 1267 sql/stats/automatic_stats.go:580  [n1] 1157  failed to get tables for automatic stats: injected error
I221120 13:29:28.199714 980 testutils/testcluster/testcluster.go:138  [-] 1158  TestCluster quiescing nodes
W221120 13:29:28.200075 1273 jobs/registry.go:840  [n1] 1159  canceling all adopted jobs due to stopper quiescing
I221120 13:29:28.200280 1267 sql/stats/automatic_stats.go:509  [n1] 1160  quiescing auto stats refresher
W221120 13:29:28.200554 1746 kv/kvserver/intentresolver/intent_resolver.go:799  [-] 1161  failed to gc transaction record: could not GC completed transaction anchored at /NamespaceTable/30/1/100/0/"public"/4/1: node unavailable; try another peer
I221120 13:29:28.201110 1575 jobs/registry.go:1253  [n1] 1162  AUTO SPAN CONFIG RECONCILIATION job 815531841817870337: stepping through state succeeded
W221120 13:29:28.201788 1256 sql/sqlliveness/slinstance/slinstance.go:307  [n1] 1163  exiting heartbeat loop
W221120 13:29:28.201956 1256 sql/sqlliveness/slinstance/slinstance.go:292  [n1] 1164  exiting heartbeat loop
W221120 13:29:28.212166 1575 jobs/adopt.go:454  [n1] 1165  could not clear job claim: clear-job-claim: failed to send RPC: sending to all replicas failed; last error: node unavailable; try another peer
I221120 13:29:28.243735 1250 kv/kvclient/kvcoord/transport_race.go:104  [-] 1166  transport race promotion: ran 41 iterations on up to 509 requests
--- FAIL: TestErrorDuringExtendedProtocolCommit (1.75s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82010 sql/tests: TestErrorDuringExtendedProtocolCommit failed [C-test-failure O-robot T-sql-experience T-sql-queries branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestErrorDuringExtendedProtocolCommit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestErrorDuringExtendedProtocolCommit [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7821692?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7821692?buildTab=artifacts#/) on master @ [ca079ac1fc8a8f7a5360f5b1f4fa210c14dc8410](https://github.com/cockroachdb/cockroach/commits/ca079ac1fc8a8f7a5360f5b1f4fa210c14dc8410):


```
I221203 14:34:57.319719 280502 13@kv/kvserver/replicate_queue.go:839  [n1,replicate,s1,r23/1:/Table/2{1-2}] 265 +trace:
I221203 14:34:57.319719 280502 13@kv/kvserver/replicate_queue.go:839  [n1,replicate,s1,r23/1:/Table/2{1-2}] 265 +     0.000ms      0.000ms    === operation:process replica _unfinished:1 _verbose:1 node:1 replicate: store:1 range:23/1:/Table/2{1-2}
I221203 14:34:57.319719 280502 13@kv/kvserver/replicate_queue.go:839  [n1,replicate,s1,r23/1:/Table/2{1-2}] 265 +     0.208ms      0.208ms    event:kv/kvserver/allocator/allocatorimpl/allocator.go:701 [n1,replicate,s1,r23/1:/Table/2{1-2}] add voter - missing voter need=3, have=1, priority=10001.00
I221203 14:34:57.319719 280502 13@kv/kvserver/replicate_queue.go:839  [n1,replicate,s1,r23/1:/Table/2{1-2}] 265 +     0.236ms      0.028ms    event:kv/kvserver/replicate_queue.go:1034 [n1,replicate,s1,r23/1:/Table/2{1-2}] next replica action: add voter
I221203 14:34:57.319719 280502 13@kv/kvserver/replicate_queue.go:839  [n1,replicate,s1,r23/1:/Table/2{1-2}] 265 +     0.317ms      0.081ms    event:kv/kvserver/allocator/allocatorimpl/allocator.go:1066 [n1,replicate,s1,r23/1:/Table/2{1-2}] allocate voter: []
I221203 14:34:57.361469 275230 jobs/registry.go:1373  [n1] 266  MIGRATION job 819225222310920193: stepping through state succeeded
I221203 14:34:57.377379 164386 jobs/wait.go:154  [n1] 267  waited for 1 [819225222310920193] queued jobs to complete 116.564168ms
I221203 14:34:57.377457 164386 upgrade/upgrademanager/manager.go:223  [n1] 268  running permanent upgrade for version 0.0-12
I221203 14:34:57.397147 164386 upgrade/upgrademanager/manager.go:522  [n1] 269  running Upgrade to 0.0-12: "create default databases"
I221203 14:34:57.397297 288446 jobs/adopt.go:245  [n1] 270  job 819225222766592001: resuming execution
I221203 14:34:57.419098 289358 jobs/registry.go:1373  [n1] 271  MIGRATION job 819225222766592001: stepping through state running
I221203 14:34:57.441901 292467 13@kv/kvserver/replicate_queue.go:839  [n1,replicate,s1,r11/1:/Table/{7-8}] 272  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I221203 14:34:57.441901 292467 13@kv/kvserver/replicate_queue.go:839  [n1,replicate,s1,r11/1:/Table/{7-8}] 272 +trace:
I221203 14:34:57.441901 292467 13@kv/kvserver/replicate_queue.go:839  [n1,replicate,s1,r11/1:/Table/{7-8}] 272 +     0.000ms      0.000ms    === operation:process replica _unfinished:1 _verbose:1 node:1 replicate: store:1 range:11/1:/Table/{7-8}
I221203 14:34:57.441901 292467 13@kv/kvserver/replicate_queue.go:839  [n1,replicate,s1,r11/1:/Table/{7-8}] 272 +     0.134ms      0.134ms    event:kv/kvserver/allocator/allocatorimpl/allocator.go:701 [n1,replicate,s1,r11/1:/Table/{7-8}] add voter - missing voter need=3, have=1, priority=10001.00
I221203 14:34:57.441901 292467 13@kv/kvserver/replicate_queue.go:839  [n1,replicate,s1,r11/1:/Table/{7-8}] 272 +     0.164ms      0.030ms    event:kv/kvserver/replicate_queue.go:1034 [n1,replicate,s1,r11/1:/Table/{7-8}] next replica action: add voter
I221203 14:34:57.441901 292467 13@kv/kvserver/replicate_queue.go:839  [n1,replicate,s1,r11/1:/Table/{7-8}] 272 +     0.240ms      0.076ms    event:kv/kvserver/allocator/allocatorimpl/allocator.go:1066 [n1,replicate,s1,r11/1:/Table/{7-8}] allocate voter: []
I221203 14:34:57.456506 292100 5@util/log/event_log.go:32  [n1,job=MIGRATION id=819225222766592001,upgrade=0.0-12,intExec=create-default-DB] 273 ={"Timestamp":1670078097436373721,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS defaultdb","Tag":"CREATE DATABASE","User":"root","DescriptorID":100,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"defaultdb"}
I221203 14:34:57.480262 294498 5@util/log/event_log.go:32  [n1,job=MIGRATION id=819225222766592001,upgrade=0.0-12,intExec=create-default-DB] 274 ={"Timestamp":1670078097458128201,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS postgres","Tag":"CREATE DATABASE","User":"root","DescriptorID":102,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"postgres"}
I221203 14:34:57.486692 297918 13@kv/kvserver/replicate_queue.go:839  [n1,replicate,s1,r16/1:/Table/1{4-5}] 275  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I221203 14:34:57.486692 297918 13@kv/kvserver/replicate_queue.go:839  [n1,replicate,s1,r16/1:/Table/1{4-5}] 275 +trace:
I221203 14:34:57.486692 297918 13@kv/kvserver/replicate_queue.go:839  [n1,replicate,s1,r16/1:/Table/1{4-5}] 275 +     0.000ms      0.000ms    === operation:process replica _unfinished:1 _verbose:1 node:1 replicate: store:1 range:16/1:/Table/1{4-5}
I221203 14:34:57.486692 297918 13@kv/kvserver/replicate_queue.go:839  [n1,replicate,s1,r16/1:/Table/1{4-5}] 275 +     0.376ms      0.376ms    event:kv/kvserver/allocator/allocatorimpl/allocator.go:701 [n1,replicate,s1,r16/1:/Table/1{4-5}] add voter - missing voter need=3, have=1, priority=10001.00
I221203 14:34:57.486692 297918 13@kv/kvserver/replicate_queue.go:839  [n1,replicate,s1,r16/1:/Table/1{4-5}] 275 +     0.404ms      0.028ms    event:kv/kvserver/replicate_queue.go:1034 [n1,replicate,s1,r16/1:/Table/1{4-5}] next replica action: add voter
I221203 14:34:57.486692 297918 13@kv/kvserver/replicate_queue.go:839  [n1,replicate,s1,r16/1:/Table/1{4-5}] 275 +     0.695ms      0.291ms    event:kv/kvserver/allocator/allocatorimpl/allocator.go:1066 [n1,replicate,s1,r16/1:/Table/1{4-5}] allocate voter: []
I221203 14:34:57.500680 289358 jobs/registry.go:1373  [n1] 276  MIGRATION job 819225222766592001: stepping through state succeeded
I221203 14:34:57.502997 299549 2@rpc/context.go:2176  [n1,rnode=1,raddr=127.0.0.1:41267,class=default,rpc] 277  connection is now ready
I221203 14:34:57.538479 164386 jobs/wait.go:154  [n1] 278  waited for 1 [819225222766592001] queued jobs to complete 141.205021ms
I221203 14:34:57.538582 164386 server/server_sql.go:1476  [n1] 279  done ensuring all necessary startup migrations have run
I221203 14:34:57.538888 302276 jobs/job_scheduler.go:433  [n1] 280  waiting 2m0s before scheduled jobs daemon start
I221203 14:34:57.539805 302411 kv/kvclient/rangefeed/rangefeedcache/watcher.go:320  [n1] 281  tenant-settings-watcher: established range feed cache
I221203 14:34:57.564444 302293 server/auto_upgrade.go:59  [n1] 282  no need to upgrade, cluster already at the newest version
I221203 14:34:57.617646 170789 3@pebble/event.go:677  [n1,s1,pebble] 283  [JOB 8] WAL created 000005
I221203 14:34:57.634421 164386 1@server/server_sql.go:1542  [n1] 284  serving sql connections
I221203 14:34:57.635650 164386 testutils/testcluster/testcluster.go:1328  [-] 285  WaitForFullReplication
I221203 14:34:57.635828 164386 testutils/testcluster/testcluster.go:1332  [-] 286  WaitForFullReplication took: 385ns
E221203 14:34:57.651558 178272 sql/stats/automatic_stats.go:580  [n1] 287  failed to get tables for automatic stats: injected error
    autocommit_extended_protocol_test.go:117: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2915/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/tests/autocommit_extended_protocol_test.go:117
        	Error:      	An error is expected but got nil.
        	Test:       	TestErrorDuringExtendedProtocolCommit
I221203 14:34:57.651944 164386 testutils/testcluster/testcluster.go:137  [-] 288  TestCluster quiescing nodes
I221203 14:34:57.652117 178272 sql/stats/automatic_stats.go:509  [n1] 289  quiescing auto stats refresher
W221203 14:34:57.652429 177831 sql/sqlliveness/slinstance/slinstance.go:312  [n1] 290  exiting heartbeat loop
W221203 14:34:57.652796 298996 kv/kvserver/intentresolver/intent_resolver.go:799  [-] 291  failed to gc transaction record: could not GC completed transaction anchored at /Table/3/1/102/2/1: node unavailable; try another peer
W221203 14:34:57.653134 179206 jobs/registry.go:953  [n1] 292  canceling all adopted jobs due to stopper quiescing
W221203 14:34:57.654025 177831 sql/sqlliveness/slinstance/slinstance.go:297  [n1] 293  exiting heartbeat loop
I221203 14:34:57.655217 204729 jobs/registry.go:1373  [n1] 294  AUTO SPAN CONFIG RECONCILIATION job 819225219980754945: stepping through state succeeded
E221203 14:34:57.655573 204729 jobs/registry.go:825  [n1] 295  error getting live session: node unavailable; try another peer
--- FAIL: TestErrorDuringExtendedProtocolCommit (1.34s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82010 sql/tests: TestErrorDuringExtendedProtocolCommit failed [C-test-failure O-robot T-sql-experience T-sql-queries branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestErrorDuringExtendedProtocolCommit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

FWIW, I stressed this on release-22.2 using a gceworker, and did eventually get a single failure after two and a half hours:

```
michae2@gceworker-michae2:~/go/src/github.com/cockroachdb/cockroach$ ./dev test pkg/sql/tests -f TestErrorDuringExtendedProtocolCommit --stress
$ bazel test //pkg/sql/tests:tests_test --test_env=GOTRACEBACK=all --test_timeout=86400 --test_env=COCKROACH_STRESS=true --run_under '@com_github_cockroachdb_stress//:stress -bazel -shardable-artifacts '"'"'XML_OUTPUT_FILE=/home/michae2/go/src/github.com/cockroachdb/cockroach/bin/dev-versions/dev.56 merge-test-xmls'"'"' ' --test_filter=TestErrorDuringExtendedProtocolCommit --test_sharding_strategy=disabled --test_output streamed
INFO: Invocation ID: f5faf693-66c4-4506-8bca-4f67b820ef57
...

88 runs so far, 0 failures, over 5s
188 runs so far, 0 failures, over 10s
283 runs so far, 0 failures, over 15s
...

179324 runs so far, 0 failures, over 2h32m30s
179424 runs so far, 0 failures, over 2h32m35s

initialized metamorphic constant "span-reuse-rate" with value 7
initialized metamorphic constant "mvcc-value-disable-simple-encoding" with value true
initialized metamorphic constant "mvcc-max-iters-before-seek" with value 2
initialized metamorphic constant "addsst-rewrite-concurrency" with value 8
initialized metamorphic constant "raft-log-truncation-clearrange-threshold" with value 24283
initialized metamorphic constant "coldata-batch-size" with value 2457
initialized metamorphic constant "kv-batch-size" with value 1
initialized metamorphic constant "datum-row-converter-batch-size" with value 1
initialized metamorphic constant "inverted-joiner-batch-size" with value 1
initialized metamorphic constant "max-batch-size" with value 4937
initialized metamorphic constant "max-batch-byte-size" with value 7273288
initialized metamorphic constant "spilling-queue-initial-len" with value 1
initialized metamorphic constant "merge-joiner-groups-buffer" with value 6
initialized metamorphic constant "ColIndexJoin-batch-size" with value 1011446
initialized metamorphic constant "ColIndexJoin-using-streamer-batch-size" with value 3838650
initialized metamorphic constant "resolve-membership-single-scan-enabled" with value false
initialized metamorphic constant "copy-batch-size" with value 494
initialized metamorphic constant "async-IE-result-channel-buffer-size" with value 19
initialized metamorphic constant "backup-merge-file-buffer-size" with value 8183973
initialized metamorphic constant "backup-read-manifest" with value true
initialized metamorphic constant "restore-worker-concurrency" with value 7
initialized metamorphic constant "changefeed.balance_range_distribution.enable" with value true
initialized metamorphic constant "changefeed.fast_gzip.enabled" with value false
initialized metamorphic constant "span-reuse-rate" with value 38
initialized metamorphic constant "mvcc-max-iters-before-seek" with value 0
initialized metamorphic constant "addsst-rewrite-concurrency" with value 15
initialized metamorphic constant "addsst-rewrite-forced" with value true
initialized metamorphic constant "kv.rangefeed.use_dedicated_connection_class.enabled" with value true
initialized metamorphic constant "coldata-batch-size" with value 1238
initialized metamorphic constant "default-batch-bytes-limit" with value 64849
initialized metamorphic constant "kv-batch-size" with value 1
initialized metamorphic constant "row-container-rows-per-chunk-shift" with value 1
initialized metamorphic constant "inverted-joiner-batch-size" with value 1
initialized metamorphic constant "max-batch-size" with value 4972
initialized metamorphic constant "max-batch-byte-size" with value 8890860
initialized metamorphic constant "parallel-scan-result-threshold" with value 4172
initialized metamorphic constant "spilling-queue-initial-len" with value 13
initialized metamorphic constant "merge-joiner-groups-buffer" with value 8
initialized metamorphic constant "ColIndexJoin-using-streamer-batch-size" with value 835947
initialized metamorphic constant "resolve-membership-single-scan-enabled" with value false
initialized metamorphic constant "use-index-lookup-for-descriptors-in-database" with value false
initialized metamorphic constant "backup-merge-file-buffer-size" with value 2768396
initialized metamorphic constant "write-metadata-sst" with value true
I221215 22:19:14.141456 1 (gostd) rand.go:176  [-] 1  random seed: 3974899377162373292
I221215 22:19:14.186156 21 3@pebble/event.go:649  [n?,pebble,s?] 2  [JOB 1] MANIFEST created 000001
I221215 22:19:14.186269 21 3@pebble/event.go:677  [n?,pebble,s?] 3  [JOB 1] WAL created 000002
I221215 22:19:14.186405 21 3@pebble/event.go:645  [n?,pebble,s?] 4  upgraded to format version: 002
I221215 22:19:14.186436 21 3@pebble/event.go:645  [n?,pebble,s?] 5  upgraded to format version: 003
I221215 22:19:14.186452 21 3@pebble/event.go:645  [n?,pebble,s?] 6  upgraded to format version: 004
I221215 22:19:14.186466 21 3@pebble/event.go:645  [n?,pebble,s?] 7  upgraded to format version: 005
I221215 22:19:14.186483 21 3@pebble/event.go:645  [n?,pebble,s?] 8  upgraded to format version: 006
I221215 22:19:14.186498 21 3@pebble/event.go:645  [n?,pebble,s?] 9  upgraded to format version: 007
I221215 22:19:14.186519 21 3@pebble/event.go:645  [n?,pebble,s?] 10  upgraded to format version: 008
I221215 22:19:14.186536 21 3@pebble/event.go:645  [n?,pebble,s?] 11  upgraded to format version: 009
I221215 22:19:14.186555 21 3@pebble/event.go:645  [n?,pebble,s?] 12  upgraded to format version: 010
I221215 22:19:14.186568 21 server/config.go:734  [n?] 13  1 storage engine initialized
I221215 22:19:14.186590 21 server/config.go:737  [n?] 14  Pebble cache size: 128 MiB
I221215 22:19:14.186610 21 server/config.go:737  [n?] 15  store 0: in-memory, size 512 MiB
I221215 22:19:14.186633 144 3@pebble/event.go:669  [n?,pebble,s?] 16  [JOB 6] all initial table stats loaded
I221215 22:19:14.188379 21 1@rpc/tls.go:268  [n?] 17  server certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I221215 22:19:14.188405 21 1@rpc/tls.go:317  [n?] 18  web UI certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I221215 22:19:14.190756 21 3@pebble/event.go:649  [n?,pebble,s?] 19  [JOB 1] MANIFEST created 000001
I221215 22:19:14.190876 21 3@pebble/event.go:677  [n?,pebble,s?] 20  [JOB 1] WAL created 000002
I221215 22:19:14.191132 195 3@pebble/event.go:669  [n?,pebble,stemp] 21  [JOB 2] all initial table stats loaded
I221215 22:19:14.237720 21 1@server/clock_monotonicity.go:65  [n?] 22  monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I221215 22:19:14.244301 21 server/init.go:197  [n?] 23  no stores initialized
I221215 22:19:14.244399 21 server/init.go:198  [n?] 24  awaiting `cockroach init` or join with an already initialized node
I221215 22:19:14.244441 21 server/init.go:262  [n?] 25  cluster d0d08cf7-f8b8-4b93-8435-b7f3fe61ae42 has been created
I221215 22:19:14.244490 21 server/init.go:263  [n?] 26  allocated node ID: n1 (for self)
I221215 22:19:14.244529 21 server/init.go:264  [n?] 27  active cluster version: 22.2
W221215 22:19:14.244781 21 1@gossip/gossip.go:1511  [n1] 28  no addresses found; use --join to specify a connected node
I221215 22:19:14.244768 342 1@server/server.go:1406  [n1] 29  connecting to gossip network to verify cluster ID "d0d08cf7-f8b8-4b93-8435-b7f3fe61ae42"
I221215 22:19:14.244860 21 gossip/gossip.go:409  [n1] 30  NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:38477" > attrs:<> locality:<tiers:<key:"region" value:"test" > tiers:<key:"dc" value:"dc1" > > ServerVersion:<major_val:22 minor_val:2 patch:0 internal:0 > build_tag:"v22.2.0-beta.2-1087-ged226a5766" started_at:1671142754244858336 cluster_name:"" sql_address:<network_field:"tcp" address_field:"127.0.0.1:37743" > http_address:<network_field:"tcp" address_field:"127.0.0.1:38035" >
I221215 22:19:14.251367 342 1@server/server.go:1409  [n1] 31  node connected via gossip
I221215 22:19:14.250842 21 server/node.go:471  [n1] 32  initialized store s1
I221215 22:19:14.251744 337 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 33  error processing replica: 0 of 0 live stores are able to take a new replica for the range (0 already have a voter, 0 already have a non-voter); likely not enough nodes in cluster
I221215 22:19:14.251744 337 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 33 +trace:
I221215 22:19:14.251744 337 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 33 +     0.000ms      0.000ms    === operation:process replica _unfinished:1 _verbose:1 node:1 replicate: store:1 range:2/1:/System/NodeLiveness{-Max}
I221215 22:19:14.251744 337 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 33 +     0.054ms      0.054ms    event:kv/kvserver/allocator/allocatorimpl/allocator.go:682 [n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] add voter - missing voter need=3, have=1, priority=10001.00
I221215 22:19:14.251744 337 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 33 +     0.069ms      0.015ms    event:kv/kvserver/replicate_queue.go:866 [n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] next replica action: add voter
I221215 22:19:14.251744 337 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 33 +     0.096ms      0.027ms    event:kv/kvserver/allocator/allocatorimpl/allocator.go:1037 [n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] allocate voter: []
I221215 22:19:14.251950 21 kv/kvserver/stores.go:262  [n1] 34  read 0 node addresses from persistent storage
I221215 22:19:14.252065 21 server/node.go:550  [n1] 36  started with engine type 2
I221215 22:19:14.252134 21 server/node.go:552  [n1] 37  started with attributes []
I221215 22:19:14.252030 483 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 35  error processing replica: 0 of 0 live stores are able to take a new replica for the range (0 already have a voter, 0 already have a non-voter); likely not enough nodes in cluster
I221215 22:19:14.252030 483 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 35 +trace:
I221215 22:19:14.252030 483 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 35 +     0.000ms      0.000ms    === operation:process replica _unfinished:1 _verbose:1 node:1 replicate: store:1 range:1/1:/{Min-System/NodeL…}
I221215 22:19:14.252030 483 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 35 +     0.032ms      0.032ms    event:kv/kvserver/allocator/allocatorimpl/allocator.go:682 [n1,replicate,s1,r1/1:/{Min-System/NodeL…}] add voter - missing voter need=3, have=1, priority=10001.00
I221215 22:19:14.252030 483 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 35 +     0.045ms      0.012ms    event:kv/kvserver/replicate_queue.go:866 [n1,replicate,s1,r1/1:/{Min-System/NodeL…}] next replica action: add voter
I221215 22:19:14.252030 483 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 35 +     0.061ms      0.016ms    event:kv/kvserver/allocator/allocatorimpl/allocator.go:1037 [n1,replicate,s1,r1/1:/{Min-System/NodeL…}] allocate voter: []
I221215 22:19:14.252255 21 1@server/server.go:1550  [n1] 38  starting https server at 127.0.0.1:38035 (use: 127.0.0.1:38035)
I221215 22:19:14.252297 21 1@server/server.go:1555  [n1] 39  starting postgres server at 127.0.0.1:37743 (use: 127.0.0.1:37743)
I221215 22:19:14.252324 21 1@server/server.go:1558  [n1] 40  starting grpc server at 127.0.0.1:38477
I221215 22:19:14.252349 21 1@server/server.go:1559  [n1] 41  advertising CockroachDB node at 127.0.0.1:38477
...

I221215 22:19:14.428191 1011 server/auto_upgrade.go:59  [n1] 93  no need to upgrade, cluster already at the newest version
I221215 22:19:14.434395 1042 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r4/1:/System{/tsd-tse}] 94  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I221215 22:19:14.434395 1042 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r4/1:/System{/tsd-tse}] 94 +trace:
I221215 22:19:14.434395 1042 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r4/1:/System{/tsd-tse}] 94 +     0.000ms      0.000ms    === operation:process replica _unfinished:1 _verbose:1 node:1 replicate: store:1 range:4/1:/System{/tsd-tse}
I221215 22:19:14.434395 1042 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r4/1:/System{/tsd-tse}] 94 +     0.038ms      0.038ms    event:kv/kvserver/allocator/allocatorimpl/allocator.go:682 [n1,replicate,s1,r4/1:/System{/tsd-tse}] add voter - missing voter need=3, have=1, priority=10001.00
I221215 22:19:14.434395 1042 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r4/1:/System{/tsd-tse}] 94 +     0.055ms      0.017ms    event:kv/kvserver/replicate_queue.go:866 [n1,replicate,s1,r4/1:/System{/tsd-tse}] next replica action: add voter
I221215 22:19:14.434395 1042 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r4/1:/System{/tsd-tse}] 94 +     0.073ms      0.019ms    event:kv/kvserver/allocator/allocatorimpl/allocator.go:1037 [n1,replicate,s1,r4/1:/System{/tsd-tse}] allocate voter: []
I221215 22:19:14.441301 406 3@pebble/event.go:677  [n1,pebble,s1] 95  [JOB 8] WAL created 000005
I221215 22:19:14.441738 983 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r50/1:/Table/5{0-1}] 96  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I221215 22:19:14.441738 983 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r50/1:/Table/5{0-1}] 96 +trace:
I221215 22:19:14.441738 983 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r50/1:/Table/5{0-1}] 96 +     0.000ms      0.000ms    === operation:process replica _unfinished:1 _verbose:1 node:1 replicate: store:1 range:50/1:/Table/5{0-1}
I221215 22:19:14.441738 983 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r50/1:/Table/5{0-1}] 96 +     0.053ms      0.053ms    event:kv/kvserver/allocator/allocatorimpl/allocator.go:682 [n1,replicate,s1,r50/1:/Table/5{0-1}] add voter - missing voter need=3, have=1, priority=10001.00
I221215 22:19:14.441738 983 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r50/1:/Table/5{0-1}] 96 +     0.071ms      0.017ms    event:kv/kvserver/replicate_queue.go:866 [n1,replicate,s1,r50/1:/Table/5{0-1}] next replica action: add voter
I221215 22:19:14.441738 983 13@kv/kvserver/replicate_queue.go:817  [n1,replicate,s1,r50/1:/Table/5{0-1}] 96 +     0.095ms      0.024ms    event:kv/kvserver/allocator/allocatorimpl/allocator.go:1037 [n1,replicate,s1,r50/1:/Table/5{0-1}] allocate voter: []
I221215 22:19:14.443372 21 1@server/server_sql.go:1451  [n1] 97  serving sql connections
I221215 22:19:14.444363 21 testutils/testcluster/testcluster.go:1339  [-] 98  WaitForFullReplication
I221215 22:19:14.444430 21 testutils/testcluster/testcluster.go:1343  [-] 99  WaitForFullReplication took: 260ns
I221215 22:19:14.444496 21 2@rpc/context.go:1774  [n1,rnode=1,raddr=127.0.0.1:38477,class=default] 100  dialing
I221215 22:19:14.468283 21 testutils/testcluster/testcluster.go:128  [-] 101  TestCluster quiescing nodes
I221215 22:19:14.468432 1055 2@rpc/context.go:1393  [n1] 102  closing [{n1: 127.0.0.1:38477 (default)} {n0: 127.0.0.1:38477 (default)}]
I221215 22:19:14.468619 537 sql/stats/automatic_stats.go:509  [n1] 103  quiescing auto stats refresher
W221215 22:19:14.468737 930 kv/kvserver/intentresolver/intent_resolver.go:799  [-] 104  failed to gc transaction record: could not GC completed transaction anchored at /Table/30/1/102/0/"public"/4/1: node unavailable; try another peer
W221215 22:19:14.468882 543 jobs/registry.go:821  [n1] 105  canceling all adopted jobs due to stopper quiescing
W221215 22:19:14.468943 45 sql/sqlliveness/slinstance/slinstance.go:242  [n1] 106  exiting heartbeat loop
I221215 22:19:14.469038 807 jobs/registry.go:1217  [n1] 107  AUTO SPAN CONFIG RECONCILIATION job 822713890462629889: stepping through state succeeded
W221215 22:19:14.469093 807 kv/txn.go:705  [n1] 108  failure aborting transaction: node unavailable; try another peer; abort caused by: context canceled
W221215 22:19:14.473481 807 jobs/adopt.go:454  [n1] 109  could not clear job claim: clear-job-claim: failed to send RPC: sending to all replicas failed; last error: node unavailable; try another peer
W221215 22:19:14.478867 977 sql/syntheticprivilegecache/cache.go:203  [n1] 110  failed to warm privileges for virtual tables: get-vtable-privileges: failed to send RPC: sending to all replicas failed; last error: node unavailable; try another peer
--- FAIL: TestErrorDuringExtendedProtocolCommit (0.37s)
    autocommit_extended_protocol_test.go:117:
                Error Trace:    /home/michae2/.cache/bazel/_bazel_michae2/88dadb79cb9eca13f5ad7e068969d8c1/sandbox/linux-sandbox/1513/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/tests/autocommit_extended_protocol_test.go:117
                Error:          An error is expected but got nil.
                Test:           TestErrorDuringExtendedProtocolCommit
FAIL
I221215 22:19:14.525027 1 (gostd) testmain.go:164  [-] 1  Test //pkg/sql/tests:tests_test exited with error code 1


ERROR: exit status 1

179505 runs completed, 1 failures, over 2h32m39s
context canceled
FAIL
FAIL: //pkg/sql/tests:tests_test (see /home/michae2/.cache/bazel/_bazel_michae2/88dadb79cb9eca13f5ad7e068969d8c1/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/testlogs/pkg/sql/tests/tests_test/test.log)
Target //pkg/sql/tests:tests_test up-to-date:
  _bazel/bin/pkg/sql/tests/tests_test_/tests_test
INFO: Elapsed time: 9171.113s, Critical Path: 9166.15s
INFO: 933 processes: 35 internal, 898 linux-sandbox.
INFO: Build completed, 1 test FAILED, 933 total actions
//pkg/sql/tests:tests_test                                               FAILED in 9159.4s
  /home/michae2/.cache/bazel/_bazel_michae2/88dadb79cb9eca13f5ad7e068969d8c1/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/testlogs/pkg/sql/tests/tests_test/test.log

INFO: Build completed, 1 test FAILED, 933 total actions
ERROR: exit status 3
```

sql/tests.TestErrorDuringExtendedProtocolCommit [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8121734?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8121734?buildTab=artifacts#/) on master @ [6723e00a46aaa3ea575093bd82a02b7d6f6b131b](https://github.com/cockroachdb/cockroach/commits/6723e00a46aaa3ea575093bd82a02b7d6f6b131b):


```
I221228 12:30:05.484417 369740 jobs/wait.go:153  [n1] 471  waited for 1 [826278560875315201] queued jobs to complete 118.708137ms
I221228 12:30:05.485470 369740 upgrade/upgrademanager/manager.go:266  [n1] 472  running permanent upgrade for version 0.0-8
I221228 12:30:05.507959 369740 upgrade/upgrademanager/manager.go:582  [n1] 473  running Upgrade to 0.0-8: "initialize the cluster.secret setting"
I221228 12:30:05.509109 466135 jobs/adopt.go:244  [n1] 474  job 826278561334984705: resuming execution
I221228 12:30:05.547915 467044 jobs/registry.go:1427  [n1] 475  MIGRATION job 826278561334984705: stepping through state running
I221228 12:30:05.591784 467044 jobs/registry.go:1427  [n1] 476  MIGRATION job 826278561334984705: stepping through state succeeded
I221228 12:30:05.618236 369740 jobs/wait.go:153  [n1] 477  waited for 1 [826278561334984705] queued jobs to complete 110.004103ms
I221228 12:30:05.619366 369740 upgrade/upgrademanager/manager.go:266  [n1] 478  running permanent upgrade for version 0.0-10
I221228 12:30:05.643539 369740 upgrade/upgrademanager/manager.go:582  [n1] 479  running Upgrade to 0.0-10: "update system.locations with default location data"
I221228 12:30:05.643691 475063 jobs/adopt.go:244  [n1] 480  job 826278561774010369: resuming execution
I221228 12:30:05.678071 475976 jobs/registry.go:1427  [n1] 481  MIGRATION job 826278561774010369: stepping through state running
I221228 12:30:05.716455 480484 13@kv/kvserver/replicate_queue.go:847  [n1,replicate,s1,r23/1:/Table/2{1-2}] 482  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I221228 12:30:05.761171 475976 jobs/registry.go:1427  [n1] 483  MIGRATION job 826278561774010369: stepping through state succeeded
I221228 12:30:05.777853 369740 jobs/wait.go:153  [n1] 484  waited for 1 [826278561774010369] queued jobs to complete 134.178151ms
I221228 12:30:05.778930 369740 upgrade/upgrademanager/manager.go:266  [n1] 485  running permanent upgrade for version 0.0-12
I221228 12:30:05.800572 369740 upgrade/upgrademanager/manager.go:582  [n1] 486  running Upgrade to 0.0-12: "create default databases"
I221228 12:30:05.800726 488559 jobs/adopt.go:244  [n1] 487  job 826278562296561665: resuming execution
I221228 12:30:05.818539 490117 13@kv/kvserver/replicate_queue.go:847  [n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 488  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I221228 12:30:05.825581 489632 jobs/registry.go:1427  [n1] 489  MIGRATION job 826278562296561665: stepping through state running
I221228 12:30:05.829510 491003 2@rpc/context.go:2297  [n1,rnode=1,raddr=127.0.0.1:38175,class=default,rpc] 490  connection is now ready
I221228 12:30:05.850858 493598 13@kv/kvserver/replicate_queue.go:847  [n1,replicate,s1,r11/1:/Table/{7-8}] 491  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I221228 12:30:05.876871 493253 5@util/log/event_log.go:32  [n1,job=MIGRATION id=826278562296561665,upgrade=0.0-12,intExec=create-default-DB] 492 ={"Timestamp":1672230605846067332,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS defaultdb","Tag":"CREATE DATABASE","User":"root","DescriptorID":100,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"defaultdb"}
I221228 12:30:05.903723 496409 5@util/log/event_log.go:32  [n1,job=MIGRATION id=826278562296561665,upgrade=0.0-12,intExec=create-default-DB] 493 ={"Timestamp":1672230605878166592,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS postgres","Tag":"CREATE DATABASE","User":"root","DescriptorID":102,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"postgres"}
I221228 12:30:05.916998 489632 jobs/registry.go:1427  [n1] 494  MIGRATION job 826278562296561665: stepping through state succeeded
I221228 12:30:05.938510 369740 jobs/wait.go:153  [n1] 495  waited for 1 [826278562296561665] queued jobs to complete 137.803577ms
I221228 12:30:05.938597 369740 server/server_sql.go:1491  [n1] 496  done ensuring all necessary startup migrations have run
I221228 12:30:05.942682 504239 kv/kvclient/rangefeed/rangefeedcache/watcher.go:320  [n1] 497  tenant-settings-watcher: established range feed cache
I221228 12:30:05.944127 503897 jobs/job_scheduler.go:408  [n1] 498  waiting 4m0s before scheduled jobs daemon start
I221228 12:30:05.982124 503935 server/auto_upgrade.go:58  [n1] 499  no need to upgrade, cluster already at the newest version
W221228 12:30:06.023709 509793 kv/kvserver/replica_rangefeed.go:323  [n1,rangefeed=tenant-settings-watcher,dest_n=1,dest_s=1,dest_r=51,s1,r51/1:/Table/5{0-1}] 500  rangefeed registration took 25.245514ms
I221228 12:30:06.039428 510982 13@kv/kvserver/replicate_queue.go:847  [n1,replicate,s1,r52/1:/Table/5{1-2}] 501  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I221228 12:30:06.084122 503923 sql/syntheticprivilegecache/cache.go:204  [n1] 502  warmed privileges for virtual tables in 143.476166ms
I221228 12:30:06.087597 381084 3@pebble/event.go:677  [n1,s1,pebble] 503  [JOB 8] WAL created 000005
I221228 12:30:06.096880 369740 1@server/server_sql.go:1552  [n1] 504  serving sql connections
I221228 12:30:06.098306 369740 testutils/testcluster/testcluster.go:1337  [-] 505  WaitForFullReplication
I221228 12:30:06.098350 369740 testutils/testcluster/testcluster.go:1341  [-] 506  WaitForFullReplication took: 271ns
E221228 12:30:06.119689 387081 sql/stats/automatic_stats.go:580  [n1] 507  failed to get tables for automatic stats: injected error
    autocommit_extended_protocol_test.go:187: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3493/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/tests/autocommit_extended_protocol_test.go:187
        	Error:      	An error is expected but got nil.
        	Test:       	TestErrorDuringExtendedProtocolCommit
I221228 12:30:06.121351 369740 testutils/testcluster/testcluster.go:146  [-] 508  TestCluster quiescing nodes
I221228 12:30:06.121543 387081 sql/stats/automatic_stats.go:509  [n1] 509  quiescing auto stats refresher
W221228 12:30:06.123063 387411 jobs/registry.go:1007  [n1] 510  canceling all adopted jobs due to stopper quiescing
W221228 12:30:06.123327 386888 sql/sqlliveness/slinstance/slinstance.go:312  [n1] 511  exiting heartbeat loop
W221228 12:30:06.122525 501730 kv/kvserver/intentresolver/intent_resolver.go:799  [-] 512  failed to gc transaction record: could not GC completed transaction anchored at /Table/3/1/102/2/1: node unavailable; try another peer
W221228 12:30:06.125136 386888 sql/sqlliveness/slinstance/slinstance.go:297  [n1] 513  exiting heartbeat loop
I221228 12:30:06.125598 415447 jobs/registry.go:1427  [n1] 514  AUTO SPAN CONFIG RECONCILIATION job 826278559280889857: stepping through state succeeded
E221228 12:30:06.126603 415447 jobs/registry.go:879  [n1] 515  error getting live session: node unavailable; try another peer
--- FAIL: TestErrorDuringExtendedProtocolCommit (1.50s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #94287 sql/tests: TestErrorDuringExtendedProtocolCommit failed [C-test-failure O-robot T-sql-queries branch-release-22.2]
- #82010 sql/tests: TestErrorDuringExtendedProtocolCommit failed [C-test-failure O-robot T-sql-queries T-sql-sessions branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestErrorDuringExtendedProtocolCommit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestErrorDuringExtendedProtocolCommit [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8243990?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8243990?buildTab=artifacts#/) on master @ [6c619e285f78a15f04a9ca75faf746ae93530f6b](https://github.com/cockroachdb/cockroach/commits/6c619e285f78a15f04a9ca75faf746ae93530f6b):


```
I230109 11:42:02.420158 472814 jobs/registry.go:1427  [T1,n1] 481  MIGRATION job 829666499923968001: stepping through state succeeded
I230109 11:42:02.437103 390170 jobs/wait.go:153  [T1,n1] 482  waited for 1 [829666499923968001] queued jobs to complete 120.45777ms
I230109 11:42:02.438227 390170 upgrade/upgrademanager/manager.go:266  [T1,n1] 483  running permanent upgrade for version 0.0-8
I230109 11:42:02.461592 390170 upgrade/upgrademanager/manager.go:582  [T1,n1] 484  running Upgrade to 0.0-8: "initialize the cluster.secret setting"
I230109 11:42:02.461743 483707 jobs/adopt.go:244  [T1,n1] 485  job 829666500405755905: resuming execution
I230109 11:42:02.481965 484595 jobs/registry.go:1427  [T1,n1] 486  MIGRATION job 829666500405755905: stepping through state running
I230109 11:42:02.520773 484595 jobs/registry.go:1427  [T1,n1] 487  MIGRATION job 829666500405755905: stepping through state succeeded
I230109 11:42:02.537574 390170 jobs/wait.go:153  [T1,n1] 488  waited for 1 [829666500405755905] queued jobs to complete 75.844597ms
I230109 11:42:02.538581 390170 upgrade/upgrademanager/manager.go:266  [T1,n1] 489  running permanent upgrade for version 0.0-10
I230109 11:42:02.559194 390170 upgrade/upgrademanager/manager.go:582  [T1,n1] 490  running Upgrade to 0.0-10: "update system.locations with default location data"
I230109 11:42:02.559339 492576 jobs/adopt.go:244  [T1,n1] 491  job 829666500734615553: resuming execution
I230109 11:42:02.582116 493482 jobs/registry.go:1427  [T1,n1] 492  MIGRATION job 829666500734615553: stepping through state running
I230109 11:42:02.617835 497947 13@kv/kvserver/replicate_queue.go:818  [T1,n1,replicate,s1,r23/1:/Table/2{1-2}] 493  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230109 11:42:02.660087 493482 jobs/registry.go:1427  [T1,n1] 494  MIGRATION job 829666500734615553: stepping through state succeeded
I230109 11:42:02.677613 390170 jobs/wait.go:153  [T1,n1] 495  waited for 1 [829666500734615553] queued jobs to complete 118.28845ms
I230109 11:42:02.678692 390170 upgrade/upgrademanager/manager.go:266  [T1,n1] 496  running permanent upgrade for version 0.0-12
I230109 11:42:02.698298 390170 upgrade/upgrademanager/manager.go:582  [T1,n1] 497  running Upgrade to 0.0-12: "create default databases"
I230109 11:42:02.698439 506049 jobs/adopt.go:244  [T1,n1] 498  job 829666501193728001: resuming execution
I230109 11:42:02.721273 507000 jobs/registry.go:1427  [T1,n1] 499  MIGRATION job 829666501193728001: stepping through state running
I230109 11:42:02.743832 510134 13@kv/kvserver/replicate_queue.go:818  [T1,n1,replicate,s1,r11/1:/Table/{7-8}] 500  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230109 11:42:02.760665 509771 5@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=829666501193728001,upgrade=0.0-12,intExec=create-default-DB] 501 ={"Timestamp":1673264522738613655,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS defaultdb","Tag":"CREATE DATABASE","User":"root","DescriptorID":100,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"defaultdb"}
I230109 11:42:02.762021 511883 2@rpc/context.go:2297  [T1,n1,rnode=1,raddr=127.0.0.1:45197,class=default,rpc] 502  connection is now ready
I230109 11:42:02.783565 512808 5@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=829666501193728001,upgrade=0.0-12,intExec=create-default-DB] 503 ={"Timestamp":1673264522762234536,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS postgres","Tag":"CREATE DATABASE","User":"root","DescriptorID":102,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"postgres"}
I230109 11:42:02.793642 507000 jobs/registry.go:1427  [T1,n1] 504  MIGRATION job 829666501193728001: stepping through state succeeded
I230109 11:42:02.818956 390170 jobs/wait.go:153  [T1,n1] 505  waited for 1 [829666501193728001] queued jobs to complete 120.534152ms
I230109 11:42:02.819127 390170 server/server_sql.go:1484  [T1,n1] 506  done ensuring all necessary startup migrations have run
I230109 11:42:02.828503 520507 jobs/job_scheduler.go:408  [T1,n1] 507  waiting 2m0s before scheduled jobs daemon start
I230109 11:42:02.831163 521709 kv/kvclient/rangefeed/rangefeedcache/watcher.go:320  [T1,n1] 508  tenant-settings-watcher: established range feed cache
I230109 11:42:02.862031 521536 server/auto_upgrade.go:78  [T1,n1] 509  no need to upgrade, cluster already at the newest version
I230109 11:42:02.872419 527292 13@kv/kvserver/replicate_queue.go:818  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 510  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230109 11:42:02.877188 521525 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 511  warmed privileges for virtual tables in 48.390233ms
I230109 11:42:02.888764 528375 13@kv/kvserver/replicate_queue.go:818  [T1,n1,replicate,s1,r4/1:/System{/tsd-tse}] 512  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230109 11:42:02.939189 396566 3@pebble/event.go:677  [T1,n1,s1,pebble] 513  [JOB 8] WAL created 000005
I230109 11:42:02.950194 390170 1@server/server_sql.go:1550  [T1,n1] 514  serving sql connections
I230109 11:42:02.951301 390170 testutils/testcluster/testcluster.go:1336  [T1] 515  WaitForFullReplication
I230109 11:42:02.961197 390170 testutils/testcluster/testcluster.go:1340  [T1] 516  WaitForFullReplication took: 584ns
E230109 11:42:02.980723 402756 sql/stats/automatic_stats.go:579  [T1,n1] 517  failed to get tables for automatic stats: injected error
    autocommit_extended_protocol_test.go:187: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3476/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/tests/autocommit_extended_protocol_test.go:187
        	Error:      	An error is expected but got nil.
        	Test:       	TestErrorDuringExtendedProtocolCommit
I230109 11:42:02.982187 390170 testutils/testcluster/testcluster.go:146  [T1] 518  TestCluster quiescing nodes
W230109 11:42:02.982994 517002 kv/kvserver/intentresolver/intent_resolver.go:799  [T1] 520  failed to gc transaction record: could not GC completed transaction anchored at /Table/3/1/102/2/1: node unavailable; try another peer
I230109 11:42:02.982354 402756 sql/stats/automatic_stats.go:508  [T1,n1] 519  quiescing auto stats refresher
W230109 11:42:02.983482 402546 sql/sqlliveness/slinstance/slinstance.go:312  [T1,n1] 521  exiting heartbeat loop
W230109 11:42:02.983619 402546 sql/sqlliveness/slinstance/slinstance.go:297  [T1,n1] 522  exiting heartbeat loop
W230109 11:42:02.983212 402795 jobs/registry.go:1007  [T1,n1] 523  canceling all adopted jobs due to stopper quiescing
I230109 11:42:02.984219 431634 jobs/registry.go:1427  [T1,n1] 524  AUTO SPAN CONFIG RECONCILIATION job 829666498313748481: stepping through state succeeded
E230109 11:42:02.984479 431634 jobs/registry.go:879  [T1,n1] 525  error getting live session: node unavailable; try another peer
--- FAIL: TestErrorDuringExtendedProtocolCommit (1.39s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82010 sql/tests: TestErrorDuringExtendedProtocolCommit failed [C-test-failure O-robot T-sql-queries T-sql-sessions branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestErrorDuringExtendedProtocolCommit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestErrorDuringExtendedProtocolCommit [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8289835?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8289835?buildTab=artifacts#/) on master @ [73485e56768d61b14a5aa9b3f880613d7820edb9](https://github.com/cockroachdb/cockroach/commits/73485e56768d61b14a5aa9b3f880613d7820edb9):


```
I230112 10:58:55.567790 467707 jobs/registry.go:1427  [T1,n1] 478  MIGRATION job 830507369920167937: stepping through state succeeded
I230112 10:58:55.584549 347817 jobs/wait.go:153  [T1,n1] 479  waited for 1 [830507369920167937] queued jobs to complete 111.631173ms
I230112 10:58:55.585775 347817 upgrade/upgrademanager/manager.go:266  [T1,n1] 480  running permanent upgrade for version 0.0-8
I230112 10:58:55.606430 347817 upgrade/upgrademanager/manager.go:582  [T1,n1] 481  running Upgrade to 0.0-8: "initialize the cluster.secret setting"
I230112 10:58:55.606572 478309 jobs/adopt.go:244  [T1,n1] 482  job 830507370367647745: resuming execution
I230112 10:58:55.628479 479208 jobs/registry.go:1427  [T1,n1] 483  MIGRATION job 830507370367647745: stepping through state running
I230112 10:58:55.673086 479208 jobs/registry.go:1427  [T1,n1] 484  MIGRATION job 830507370367647745: stepping through state succeeded
I230112 10:58:55.688842 347817 jobs/wait.go:153  [T1,n1] 485  waited for 1 [830507370367647745] queued jobs to complete 82.282791ms
I230112 10:58:55.689659 347817 upgrade/upgrademanager/manager.go:266  [T1,n1] 486  running permanent upgrade for version 0.0-10
I230112 10:58:55.709585 347817 upgrade/upgrademanager/manager.go:582  [T1,n1] 487  running Upgrade to 0.0-10: "update system.locations with default location data"
I230112 10:58:55.709737 487247 jobs/adopt.go:244  [T1,n1] 488  job 830507370708598785: resuming execution
I230112 10:58:55.731632 488156 jobs/registry.go:1427  [T1,n1] 489  MIGRATION job 830507370708598785: stepping through state running
I230112 10:58:55.767452 492484 13@kv/kvserver/replicate_queue.go:818  [T1,n1,replicate,s1,r23/1:/Table/2{1-2}] 490  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230112 10:58:55.812617 488156 jobs/registry.go:1427  [T1,n1] 491  MIGRATION job 830507370708598785: stepping through state succeeded
I230112 10:58:55.829353 347817 jobs/wait.go:153  [T1,n1] 492  waited for 1 [830507370708598785] queued jobs to complete 119.630218ms
I230112 10:58:55.830493 347817 upgrade/upgrademanager/manager.go:266  [T1,n1] 493  running permanent upgrade for version 0.0-12
I230112 10:58:55.850499 347817 upgrade/upgrademanager/manager.go:582  [T1,n1] 494  running Upgrade to 0.0-12: "create default databases"
I230112 10:58:55.850653 500580 jobs/adopt.go:244  [T1,n1] 495  job 830507371169579009: resuming execution
I230112 10:58:55.859080 501428 13@kv/kvserver/replicate_queue.go:818  [T1,n1,replicate,s1,r24/1:/Table/2{2-3}] 496  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230112 10:58:55.868342 502191 2@rpc/context.go:2297  [T1,n1,rnode=1,raddr=127.0.0.1:37203,class=default,rpc] 497  connection is now ready
I230112 10:58:55.876545 502306 jobs/registry.go:1427  [T1,n1] 498  MIGRATION job 830507371169579009: stepping through state running
I230112 10:58:55.901451 505636 13@kv/kvserver/replicate_queue.go:818  [T1,n1,replicate,s1,r11/1:/Table/{7-8}] 499  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230112 10:58:55.917893 505218 5@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=830507371169579009,upgrade=0.0-12,intExec=create-default-DB] 500 ={"Timestamp":1673521135895305942,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS defaultdb","Tag":"CREATE DATABASE","User":"root","DescriptorID":100,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"defaultdb"}
I230112 10:58:55.943998 507699 5@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=830507371169579009,upgrade=0.0-12,intExec=create-default-DB] 501 ={"Timestamp":1673521135920471912,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS postgres","Tag":"CREATE DATABASE","User":"root","DescriptorID":102,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"postgres"}
I230112 10:58:55.964034 502306 jobs/registry.go:1427  [T1,n1] 502  MIGRATION job 830507371169579009: stepping through state succeeded
I230112 10:58:55.987854 347817 jobs/wait.go:153  [T1,n1] 503  waited for 1 [830507371169579009] queued jobs to complete 137.213412ms
I230112 10:58:55.987954 347817 server/server_sql.go:1484  [T1,n1] 504  done ensuring all necessary startup migrations have run
I230112 10:58:55.988177 515877 jobs/job_scheduler.go:408  [T1,n1] 505  waiting 3m0s before scheduled jobs daemon start
I230112 10:58:55.990531 516324 kv/kvclient/rangefeed/rangefeedcache/watcher.go:320  [T1,n1] 506  tenant-settings-watcher: established range feed cache
I230112 10:58:56.020294 516029 server/auto_upgrade.go:78  [T1,n1] 507  no need to upgrade, cluster already at the newest version
I230112 10:58:56.057557 522340 13@kv/kvserver/replicate_queue.go:818  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 508  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230112 10:58:56.062739 515977 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 509  warmed privileges for virtual tables in 73.717513ms
I230112 10:58:56.102463 390471 3@pebble/event.go:677  [T1,n1,s1,pebble] 510  [JOB 8] WAL created 000005
I230112 10:58:56.108072 347817 1@server/server_sql.go:1550  [T1,n1] 511  serving sql connections
I230112 10:58:56.109218 347817 testutils/testcluster/testcluster.go:1336  [T1] 512  WaitForFullReplication
I230112 10:58:56.109260 347817 testutils/testcluster/testcluster.go:1340  [T1] 513  WaitForFullReplication took: 308ns
E230112 10:58:56.123263 398853 sql/stats/automatic_stats.go:579  [T1,n1] 514  failed to get tables for automatic stats: injected error
    autocommit_extended_protocol_test.go:187: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2653/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/tests/autocommit_extended_protocol_test.go:187
        	Error:      	An error is expected but got nil.
        	Test:       	TestErrorDuringExtendedProtocolCommit
I230112 10:58:56.124092 347817 testutils/testcluster/testcluster.go:146  [T1] 515  TestCluster quiescing nodes
I230112 10:58:56.124281 398853 sql/stats/automatic_stats.go:508  [T1,n1] 516  quiescing auto stats refresher
W230112 10:58:56.124781 512486 kv/kvserver/intentresolver/intent_resolver.go:799  [T1] 517  failed to gc transaction record: could not GC completed transaction anchored at /Table/3/1/102/2/1: node unavailable; try another peer
W230112 10:58:56.125025 398974 jobs/registry.go:1007  [T1,n1] 518  canceling all adopted jobs due to stopper quiescing
W230112 10:58:56.125751 398614 sql/sqlliveness/slinstance/slinstance.go:312  [T1,n1] 519  exiting heartbeat loop
W230112 10:58:56.126299 398614 sql/sqlliveness/slinstance/slinstance.go:297  [T1,n1] 520  exiting heartbeat loop
I230112 10:58:56.126892 426328 jobs/registry.go:1427  [T1,n1] 521  AUTO SPAN CONFIG RECONCILIATION job 830507368264400897: stepping through state succeeded
E230112 10:58:56.127064 426328 jobs/registry.go:879  [T1,n1] 522  error getting live session: node unavailable; try another peer
--- FAIL: TestErrorDuringExtendedProtocolCommit (1.42s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82010 sql/tests: TestErrorDuringExtendedProtocolCommit failed [C-test-failure O-robot T-sql-queries T-sql-sessions branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestErrorDuringExtendedProtocolCommit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestErrorDuringExtendedProtocolCommit [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9015961?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9015961?buildTab=artifacts#/) on master @ [5b2a5670cbbe895d76602c230390816e783e0caa](https://github.com/cockroachdb/cockroach/commits/5b2a5670cbbe895d76602c230390816e783e0caa):


```
I230311 13:17:30.463340 4475 2@rpc/context.go:2511  [T1,n1,rnode=1,raddr=127.0.0.1:37007,class=default,rpc] 588  connection is now ready
I230311 13:17:30.469687 4514 jobs/registry.go:1495  [T1,n1] 589  MIGRATION job 846955316338163713: stepping through state succeeded
I230311 13:17:30.490439 3366 jobs/wait.go:142  [T1,n1] 590  waited for 1 [846955316338163713] queued jobs to complete 118.178693ms
I230311 13:17:30.491294 3366 upgrade/upgrademanager/manager.go:268  [T1,n1] 591  running permanent upgrade for version 1000022.2-32
I230311 13:17:30.514967 3366 upgrade/upgrademanager/manager.go:584  [T1,n1] 592  running Upgrade to 1000022.2-32: "add tables and jobs to support persisting key visualizer samples"
I230311 13:17:30.515231 4599 jobs/adopt.go:247  [T1,n1] 593  job 846955316810416129: resuming execution
I230311 13:17:30.536760 4556 jobs/registry.go:1495  [T1,n1] 594  MIGRATION job 846955316810416129: stepping through state running
I230311 13:17:30.570893 4560 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=846955316810416129,upgrade=1000022.2-32,intExec=set-SpanStatsTenantBoundariesTable-TTL] 595 ={"Timestamp":1678540650559902074,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.span_stats_tenant_boundaries CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":57,"ApplicationName":"$ internal-set-SpanStatsTenantBoundariesTable-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.span_stats_tenant_boundaries","Options":["\"gc.ttlseconds\" = 3600"]}
I230311 13:17:30.628456 4556 jobs/registry.go:1495  [T1,n1] 596  MIGRATION job 846955316810416129: stepping through state succeeded
I230311 13:17:30.660533 3366 jobs/wait.go:142  [T1,n1] 597  waited for 1 [846955316810416129] queued jobs to complete 145.345427ms
I230311 13:17:30.662066 3366 upgrade/upgrademanager/manager.go:268  [T1,n1] 598  running permanent upgrade for version 1000022.2-38
I230311 13:17:30.692540 3366 upgrade/upgrademanager/manager.go:584  [T1,n1] 599  running Upgrade to 1000022.2-38: "create jobs metrics polling job"
I230311 13:17:30.692988 4632 jobs/adopt.go:247  [T1,n1] 600  job 846955317370355713: resuming execution
I230311 13:17:30.725714 4581 jobs/registry.go:1495  [T1,n1] 601  MIGRATION job 846955317370355713: stepping through state running
I230311 13:17:30.772655 4581 jobs/registry.go:1495  [T1,n1] 602  MIGRATION job 846955317370355713: stepping through state succeeded
I230311 13:17:30.793802 3635 3@pebble/event.go:694  [T1,n1,s1,pebble] 603  [JOB 6] WAL created 000005
I230311 13:17:30.796746 3366 jobs/wait.go:142  [T1,n1] 604  waited for 1 [846955317370355713] queued jobs to complete 103.882438ms
I230311 13:17:30.796942 3366 server/server_sql.go:1540  [T1,n1] 605  done ensuring all necessary startup migrations have run
I230311 13:17:30.798418 4527 jobs/job_scheduler.go:407  [T1,n1] 606  waiting 4m0s before scheduled jobs daemon start
I230311 13:17:30.800464 4693 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 607  tenant-settings-watcher: established range feed cache
I230311 13:17:30.803978 4665 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 608  tenant-capability-watcher: established range feed cache
I230311 13:17:30.810691 4680 13@kv/kvserver/replicate_queue.go:819  [T1,n1,replicate,s1,r58/1:/{Table/57-Max}] 609  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230311 13:17:30.812322 4683 13@kv/kvserver/replicate_queue.go:819  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 610  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230311 13:17:30.814535 4665 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:147  [T1,n1] 611  received results of a full table scan for tenant capabilities
I230311 13:17:30.823638 4691 server/auto_upgrade.go:78  [T1,n1] 612  no need to upgrade, cluster already at the newest version
I230311 13:17:30.831274 4756 13@kv/kvserver/replicate_queue.go:819  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 613  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230311 13:17:30.838615 4529 sql/syntheticprivilegecache/cache.go:195  [T1,n1] 614  warmed privileges for virtual tables in 40.075774ms
I230311 13:17:31.018036 4767 13@kv/kvserver/replicate_queue.go:819  [T1,n1,replicate,s1,r4/1:/System{/tsd-tse}] 615  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230311 13:17:31.443636 4780 13@kv/kvserver/replicate_queue.go:819  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 616  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230311 13:17:31.644705 3602 1@gossip/gossip.go:1420  [T1,n1] 617  node has connected to cluster via gossip
I230311 13:17:31.645318 3602 kv/kvserver/stores.go:282  [T1,n1] 618  wrote 0 node addresses to persistent storage
I230311 13:17:31.795860 3366 1@server/server_sql.go:1627  [T1,n1] 619  serving sql connections
I230311 13:17:31.803214 3366 testutils/testcluster/testcluster.go:1338  [T1] 620  WaitForFullReplication
I230311 13:17:31.803322 3366 testutils/testcluster/testcluster.go:1342  [T1] 621  WaitForFullReplication took: 837ns
    autocommit_extended_protocol_test.go:187: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3984/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/tests/autocommit_extended_protocol_test.go:187
        	Error:      	An error is expected but got nil.
        	Test:       	TestErrorDuringExtendedProtocolCommit
I230311 13:17:31.844163 3366 testutils/testcluster/testcluster.go:148  [T1] 622  TestCluster quiescing nodes
W230311 13:17:31.844491 3982 jobs/registry.go:1080  [T1,n1] 623  canceling all adopted jobs due to stopper quiescing
I230311 13:17:31.845043 3405 server/start_listen.go:103  [T1,n1] 624  server shutting down: instructing cmux to stop accepting
I230311 13:17:31.845233 3976 sql/stats/automatic_stats.go:507  [T1,n1] 626  quiescing auto stats refresher
I230311 13:17:31.845197 3361 kv/kvclient/kvcoord/transport_race.go:104  [T1] 625  transport race promotion: ran 34 iterations on up to 697 requests
I230311 13:17:31.847147 4475 rpc/context.go:2249  [T1,n1,rnode=1,raddr=127.0.0.1:37007,class=default,rpc] 627  connection heartbeat loop ended with err: <nil>
W230311 13:17:31.847475 3840 sql/sqlliveness/slinstance/slinstance.go:312  [T1,n1] 628  exiting heartbeat loop
W230311 13:17:31.847644 3840 sql/sqlliveness/slinstance/slinstance.go:297  [T1,n1] 629  exiting heartbeat loop
I230311 13:17:31.847759 4096 jobs/registry.go:1495  [T1,n1] 630  AUTO SPAN CONFIG RECONCILIATION job 846955313764532225: stepping through state succeeded
E230311 13:17:31.847810 3840 server/server_sql.go:477  [T1,n1] 631  failed to run update of instance with new session ID: node unavailable; try another peer
E230311 13:17:31.848246 4096 jobs/registry.go:952  [T1,n1] 632  error getting live session: node unavailable; try another peer
--- FAIL: TestErrorDuringExtendedProtocolCommit (2.88s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82010 sql/tests: TestErrorDuringExtendedProtocolCommit failed [C-test-failure O-robot T-sql-queries T-sql-sessions branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestErrorDuringExtendedProtocolCommit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestErrorDuringExtendedProtocolCommit [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9135462?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9135462?buildTab=artifacts#/) on master @ [53dbb86acb1d48309530181b94838faf937084d3](https://github.com/cockroachdb/cockroach/commits/53dbb86acb1d48309530181b94838faf937084d3):


```
I230319 12:10:07.034366 8814 jobs/adopt.go:261  [T1,n1] 611  job 849206991182921729: resuming execution
I230319 12:10:07.084052 8866 jobs/registry.go:1494  [T1,n1] 612  MIGRATION job 849206991182921729: stepping through state running
I230319 12:10:07.137675 8916 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=849206991182921729,upgrade=1000022.2-32,intExec=set-SpanStatsTenantBoundariesTable-TTL] 613 ={"Timestamp":1679227807127367959,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.span_stats_tenant_boundaries CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":57,"ApplicationName":"$ internal-set-SpanStatsTenantBoundariesTable-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.span_stats_tenant_boundaries","Options":["\"gc.ttlseconds\" = 3600"]}
I230319 12:10:07.182696 8866 jobs/registry.go:1494  [T1,n1] 614  MIGRATION job 849206991182921729: stepping through state succeeded
I230319 12:10:07.257088 7328 jobs/wait.go:142  [T1,n1] 615  waited for 1 [849206991182921729] queued jobs to complete 222.806735ms
I230319 12:10:07.258051 7328 upgrade/upgrademanager/manager.go:268  [T1,n1] 616  running permanent upgrade for version 1000022.2-38
I230319 12:10:07.315252 7328 upgrade/upgrademanager/manager.go:585  [T1,n1] 617  running Upgrade to 1000022.2-38: "create jobs metrics polling job"
I230319 12:10:07.315614 8969 jobs/adopt.go:261  [T1,n1] 618  job 849206992103735297: resuming execution
I230319 12:10:07.369970 8936 jobs/registry.go:1494  [T1,n1] 619  MIGRATION job 849206992103735297: stepping through state running
I230319 12:10:07.452059 8936 jobs/registry.go:1494  [T1,n1] 620  MIGRATION job 849206992103735297: stepping through state succeeded
I230319 12:10:07.520763 7328 jobs/wait.go:142  [T1,n1] 621  waited for 1 [849206992103735297] queued jobs to complete 205.232083ms
I230319 12:10:07.521632 7328 upgrade/upgrademanager/manager.go:268  [T1,n1] 622  running permanent upgrade for version 1000022.2-90
I230319 12:10:07.574789 7328 upgrade/upgrademanager/manager.go:585  [T1,n1] 623  running Upgrade to 1000022.2-90: "create auto config runner job"
I230319 12:10:07.575142 9021 jobs/adopt.go:261  [T1,n1] 624  job 849206992967532545: resuming execution
I230319 12:10:07.625004 9033 jobs/registry.go:1494  [T1,n1] 625  MIGRATION job 849206992967532545: stepping through state running
I230319 12:10:07.715479 9079 jobs/adopt.go:261  [T1,n1,job=MIGRATION id=849206992967532545,upgrade=1000022.2-90] 626  job 102: resuming execution
I230319 12:10:07.723525 9033 jobs/registry.go:1494  [T1,n1] 627  MIGRATION job 849206992967532545: stepping through state succeeded
E230319 12:10:07.737644 9079 jobs/adopt.go:225  [T1,n1,job=MIGRATION id=849206992967532545,upgrade=1000022.2-90] 628  could not run claimed job 102: no resumer is available for AUTO CONFIG RUNNER
I230319 12:10:07.792111 7328 jobs/wait.go:142  [T1,n1] 629  waited for 1 [849206992967532545] queued jobs to complete 217.038069ms
I230319 12:10:07.792280 7328 server/server_sql.go:1557  [T1,n1] 630  done ensuring all necessary startup migrations have run
I230319 12:10:07.793489 9089 jobs/job_scheduler.go:407  [T1,n1] 631  waiting 4m0s before scheduled jobs daemon start
I230319 12:10:07.795320 9127 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 632  tenant-settings-watcher: established range feed cache
I230319 12:10:07.797438 9129 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 633  tenant-capability-watcher: established range feed cache
I230319 12:10:07.804069 9143 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 634  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230319 12:10:07.805817 9129 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:148  [T1,n1] 635  received results of a full table scan for tenant capabilities
I230319 12:10:07.806272 9145 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 636  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230319 12:10:07.818906 9125 server/auto_upgrade.go:78  [T1,n1] 637  no need to upgrade, cluster already at the newest version
I230319 12:10:07.826020 9150 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 638  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230319 12:10:07.834171 9123 sql/syntheticprivilegecache/cache.go:195  [T1,n1] 639  warmed privileges for virtual tables in 40.677067ms
I230319 12:10:07.999730 8897 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r2/1:/System/NodeLiveness{-Max}] 640  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230319 12:10:08.778156 7610 3@pebble/event.go:734  [n1,s1,pebble] 641  [JOB 7] WAL created 000006
I230319 12:10:08.794786 7328 1@server/server_sql.go:1644  [T1,n1] 642  serving sql connections
I230319 12:10:08.801052 7328 testutils/testcluster/testcluster.go:1352  [-] 643  WaitForFullReplication
I230319 12:10:08.801176 7328 testutils/testcluster/testcluster.go:1356  [-] 644  WaitForFullReplication took: 666ns
    autocommit_extended_protocol_test.go:187: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/tests/autocommit_extended_protocol_test.go:187
        	Error:      	An error is expected but got nil.
        	Test:       	TestErrorDuringExtendedProtocolCommit
I230319 12:10:08.838512 7328 testutils/testcluster/testcluster.go:148  [-] 645  TestCluster quiescing nodes
I230319 12:10:08.839360 7710 kv/kvclient/kvcoord/transport_race.go:104  [-] 646  transport race promotion: ran 38 iterations on up to 1118 requests
W230319 12:10:08.839452 7951 jobs/registry.go:1052  [T1,n1] 647  canceling all adopted jobs due to stopper quiescing
I230319 12:10:08.839446 7945 sql/stats/automatic_stats.go:507  [T1,n1] 648  quiescing auto stats refresher
I230319 12:10:08.840266 7561 server/start_listen.go:103  [T1,n1] 649  server shutting down: instructing cmux to stop accepting
W230319 12:10:08.841018 7814 sql/sqlliveness/slinstance/slinstance.go:312  [n1] 650  exiting heartbeat loop
W230319 12:10:08.841162 7814 sql/sqlliveness/slinstance/slinstance.go:297  [n1] 651  exiting heartbeat loop
E230319 12:10:08.841277 7814 server/server_sql.go:490  [n1] 652  failed to run update of instance with new session ID: node unavailable; try another peer
I230319 12:10:08.841416 8414 rpc/context.go:2249  [T1,n1,rnode=1,raddr=127.0.0.1:38877,class=default,rpc] 653  connection heartbeat loop ended with err: <nil>
I230319 12:10:08.841902 8044 jobs/registry.go:1494  [T1,n1] 654  AUTO SPAN CONFIG RECONCILIATION job 849206985215508481: stepping through state succeeded
E230319 12:10:08.842241 8044 jobs/registry.go:924  [T1,n1] 655  error getting live session: node unavailable; try another peer
--- FAIL: TestErrorDuringExtendedProtocolCommit (4.32s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82010 sql/tests: TestErrorDuringExtendedProtocolCommit failed [C-test-failure O-robot T-sql-queries T-sql-sessions branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestErrorDuringExtendedProtocolCommit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestErrorDuringExtendedProtocolCommit [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9145916?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9145916?buildTab=artifacts#/) on master @ [6839729bc949ad8197319df2371111e8918d89de](https://github.com/cockroachdb/cockroach/commits/6839729bc949ad8197319df2371111e8918d89de):


```
I230320 12:14:49.278178 6534 jobs/wait.go:142  [T1,n1] 601  waited for 1 [849491031496884225] queued jobs to complete 63.295643ms
I230320 12:14:49.278344 6534 upgrade/upgrademanager/manager.go:270  [T1,n1] 602  running permanent upgrade for version 1000022.2-38
I230320 12:14:49.289611 6534 upgrade/upgrademanager/manager.go:708  [T1,n1] 603  running Upgrade to 1000022.2-38: "create jobs metrics polling job"
I230320 12:14:49.289689 7967 jobs/adopt.go:261  [T1,n1] 604  job 849491031747395585: resuming execution
I230320 12:14:49.307018 7986 jobs/registry.go:1494  [T1,n1] 605  MIGRATION job 849491031747395585: stepping through state running
I230320 12:14:49.333068 7986 jobs/registry.go:1494  [T1,n1] 606  MIGRATION job 849491031747395585: stepping through state succeeded
I230320 12:14:49.357354 6534 jobs/wait.go:142  [T1,n1] 607  waited for 1 [849491031747395585] queued jobs to complete 67.680814ms
I230320 12:14:49.357549 6534 upgrade/upgrademanager/manager.go:270  [T1,n1] 608  running permanent upgrade for version 1000022.2-90
I230320 12:14:49.369832 6534 upgrade/upgrademanager/manager.go:708  [T1,n1] 609  running Upgrade to 1000022.2-90: "create auto config runner job"
I230320 12:14:49.369888 8055 jobs/adopt.go:261  [T1,n1] 610  job 849491032006918145: resuming execution
I230320 12:14:49.385912 8030 jobs/registry.go:1494  [T1,n1] 611  MIGRATION job 849491032006918145: stepping through state running
I230320 12:14:49.414208 8030 jobs/registry.go:1494  [T1,n1] 613  MIGRATION job 849491032006918145: stepping through state succeeded
I230320 12:14:49.414196 8106 jobs/adopt.go:261  [T1,n1,job=MIGRATION id=849491032006918145,upgrade=1000022.2-90] 612  job 102: resuming execution
E230320 12:14:49.422000 8106 jobs/adopt.go:225  [T1,n1,job=MIGRATION id=849491032006918145,upgrade=1000022.2-90] 614  could not run claimed job 102: no resumer is available for AUTO CONFIG RUNNER
I230320 12:14:49.445662 6534 jobs/wait.go:142  [T1,n1] 615  waited for 1 [849491032006918145] queued jobs to complete 75.782968ms
I230320 12:14:49.445718 6534 server/server_sql.go:1558  [T1,n1] 616  done ensuring all necessary startup migrations have run
I230320 12:14:49.446024 8123 jobs/job_scheduler.go:407  [T1,n1] 617  waiting 3m0s before scheduled jobs daemon start
I230320 12:14:49.446351 8129 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 618  tenant-settings-watcher: established range feed cache
I230320 12:14:49.450032 8131 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 619  tenant-capability-watcher: established range feed cache
I230320 12:14:49.452740 8127 server/auto_upgrade.go:78  [T1,n1] 620  no need to upgrade, cluster already at the newest version
I230320 12:14:49.453920 8178 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 621  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230320 12:14:49.454616 8131 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:148  [T1,n1] 622  received results of a full table scan for tenant capabilities
I230320 12:14:49.454813 8125 sql/syntheticprivilegecache/cache.go:195  [T1,n1] 623  warmed privileges for virtual tables in 8.701408ms
I230320 12:14:49.454891 8181 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 624  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230320 12:14:49.455087 8183 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 625  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230320 12:14:49.506917 8211 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r4/1:/System{/tsd-tse}] 626  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230320 12:14:49.517618 6790 3@pebble/event.go:734  [n1,s1,pebble] 627  [JOB 7] WAL created 000006
I230320 12:14:49.520466 6534 1@server/server_sql.go:1645  [T1,n1] 628  serving sql connections
I230320 12:14:49.521430 6534 testutils/testcluster/testcluster.go:1352  [-] 629  WaitForFullReplication
I230320 12:14:49.521463 6534 testutils/testcluster/testcluster.go:1356  [-] 630  WaitForFullReplication took: 259ns
I230320 12:14:49.539579 8260 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r6/1:/Table/{0-3}] 631  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230320 12:14:49.542296 8245 2@rpc/context.go:2519  [T1,n1,rnode=1,raddr=127.0.0.1:42323,class=default,rpc] 632  connection is now ready
    autocommit_extended_protocol_test.go:187: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/tests/autocommit_extended_protocol_test.go:187
        	Error:      	An error is expected but got nil.
        	Test:       	TestErrorDuringExtendedProtocolCommit
I230320 12:14:49.557090 6534 testutils/testcluster/testcluster.go:148  [-] 633  TestCluster quiescing nodes
W230320 12:14:49.558237 8096 kv/kvserver/intentresolver/intent_resolver.go:826  [-] 634  failed to gc transaction record: could not GC completed transaction anchored at /Table/15/1/849491032006918145/0: node unavailable; try another peer
W230320 12:14:49.560003 7049 jobs/registry.go:1052  [T1,n1] 635  canceling all adopted jobs due to stopper quiescing
I230320 12:14:49.560314 6593 server/start_listen.go:103  [T1,n1] 636  server shutting down: instructing cmux to stop accepting
I230320 12:14:49.561126 7256 jobs/registry.go:1494  [T1,n1] 637  AUTO SPAN CONFIG RECONCILIATION job 849491029375352833: stepping through state succeeded
E230320 12:14:49.561750 7043 sql/stats/automatic_stats.go:578  [T1,n1] 638  failed to get tables for automatic stats: get-tables: node unavailable; try another peer
I230320 12:14:49.562610 7043 sql/stats/automatic_stats.go:507  [T1,n1] 641  quiescing auto stats refresher
W230320 12:14:49.562292 6716 sql/sqlliveness/slinstance/slinstance.go:312  [n1] 639  exiting heartbeat loop
W230320 12:14:49.562757 6716 sql/sqlliveness/slinstance/slinstance.go:297  [n1] 643  exiting heartbeat loop
E230320 12:14:49.562831 6716 server/server_sql.go:493  [n1] 644  failed to run update of instance with new session ID: node unavailable; try another peer
E230320 12:14:49.562578 8266 kv/kvclient/kvcoord/txn_interceptor_committer.go:489  [-] 642  making txn commit explicit failed for "unnamed" meta={id=49f8eeb4 key=/Table/11/2/"\x80"/3/1/2023-03-20T12:20:04.548979Z/1/0 pri=100.00000000 epo=0 ts=1679314489.552991107,0 min=1679314489.552991107,0 seq=2} lock=true stat=STAGING rts=1679314489.552991107,0 wto=false gul=1679314490.052991107,0 ifw=1: result is ambiguous: error=ba: EndTxn(commit) [/Table/11/2/"\x80"/3/1/2023-03-20T12:20:04.548979Z/1/0], [txn: 49f8eeb4] RPC error: node unavailable; try another peer [exhausted]
I230320 12:14:49.562487 8245 rpc/context.go:2257  [T1,n1,rnode=1,raddr=127.0.0.1:42323,class=default,rpc] 640  connection heartbeat loop ended with err: <nil>
W230320 12:14:49.567528 7256 jobs/adopt.go:511  [T1,n1] 645  could not clear job claim: clear-job-claim: failed to send RPC: sending to all replicas failed; last error: ba: Scan [/Table/15/1/849491029375352833,/Table/15/1/849491029375352834), [txn: 0feb4be1], [can-forward-ts] RPC error: node unavailable; try another peer
--- FAIL: TestErrorDuringExtendedProtocolCommit (1.18s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82010 sql/tests: TestErrorDuringExtendedProtocolCommit failed [C-test-failure O-robot T-sql-queries T-sql-sessions branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestErrorDuringExtendedProtocolCommit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestErrorDuringExtendedProtocolCommit [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9388481?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9388481?buildTab=artifacts#/) on master @ [1e27c65fb5a3810a5e2285cf07b45d95d401cca6](https://github.com/cockroachdb/cockroach/commits/1e27c65fb5a3810a5e2285cf07b45d95d401cca6):


```
I230402 14:00:00.045635 7751 upgrade/upgrademanager/manager.go:270  [T1,n1] 690  running permanent upgrade for version 1000022.2-94
I230402 14:00:00.064953 7751 jobs/registry.go:617  [T1,n1] 691  active version is 1000023.1-4
I230402 14:00:00.074229 7751 upgrade/upgrademanager/manager.go:708  [T1,n1] 692  running Upgrade to 1000022.2-94: "create statement_activity and transaction_activity tables"
I230402 14:00:00.074338 9558 jobs/adopt.go:261  [T1,n1] 693  job 853192212629716993: resuming execution
I230402 14:00:00.086858 9563 jobs/registry.go:1522  [T1,n1] 694  MIGRATION job 853192212629716993: stepping through state running
I230402 14:00:00.095896 9563 jobs/registry.go:1522  [T1,n1] 695  MIGRATION job 853192212629716993: stepping through state succeeded
I230402 14:00:00.114276 8824 2@rpc/context.go:2565  [T1,n1,rnode=1,raddr=127.0.0.1:38585,class=default,rpc] 696  connection is now ready
I230402 14:00:00.116609 7751 jobs/wait.go:145  [T1,n1] 697  waited for 1 [853192212629716993] queued jobs to complete 42.315581ms
I230402 14:00:00.116809 7751 upgrade/upgrademanager/manager.go:270  [T1,n1] 698  running permanent upgrade for version 1000022.2-96
I230402 14:00:00.138874 7751 jobs/registry.go:617  [T1,n1] 699  active version is 1000023.1-4
I230402 14:00:00.147103 7751 upgrade/upgrademanager/manager.go:708  [T1,n1] 700  running Upgrade to 1000022.2-96: "change TTL for SQL Stats system tables"
I230402 14:00:00.147200 9612 jobs/adopt.go:261  [T1,n1] 701  job 853192212862894081: resuming execution
I230402 14:00:00.161012 9617 jobs/registry.go:1522  [T1,n1] 702  MIGRATION job 853192212862894081: stepping through state running
I230402 14:00:00.177666 9675 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=853192212862894081,upgrade=1000022.2-96,intExec=set-SQLStatsTables-TTL] 703 ={"Timestamp":1680444000170620441,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230402 14:00:00.185271 9735 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=853192212862894081,upgrade=1000022.2-96,intExec=set-SQLStatsTables-TTL] 704 ={"Timestamp":1680444000180602837,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230402 14:00:00.188665 9747 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=853192212862894081,upgrade=1000022.2-96,intExec=set-SQLStatsTables-TTL] 705 ={"Timestamp":1680444000185627450,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230402 14:00:00.191839 9752 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=853192212862894081,upgrade=1000022.2-96,intExec=set-SQLStatsTables-TTL] 706 ={"Timestamp":1680444000189009403,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230402 14:00:00.193714 9617 jobs/registry.go:1522  [T1,n1] 707  MIGRATION job 853192212862894081: stepping through state succeeded
I230402 14:00:00.219346 7751 jobs/wait.go:145  [T1,n1] 708  waited for 1 [853192212862894081] queued jobs to complete 72.190058ms
I230402 14:00:00.219444 7751 server/server_sql.go:1587  [T1,n1] 709  done ensuring all necessary startup migrations have run
I230402 14:00:00.220570 9387 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 710  tenant-settings-watcher: established range feed cache
I230402 14:00:00.222639 9810 jobs/job_scheduler.go:407  [T1,n1] 711  waiting 2m0s before scheduled jobs daemon start
I230402 14:00:00.223009 9389 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 712  tenant-capability-watcher: established range feed cache
I230402 14:00:00.226453 9844 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 713  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230402 14:00:00.226501 9389 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 714  received results of a full table scan for tenant capabilities
I230402 14:00:00.227433 9864 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r58/1:/Table/5{7-8}] 715  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230402 14:00:00.228375 9385 server/auto_upgrade.go:78  [T1,n1] 716  no need to upgrade, cluster already at the newest version
I230402 14:00:00.230324 9659 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 717  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230402 14:00:00.231512 9812 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 718  warmed privileges for virtual tables in 11.581297ms
I230402 14:00:00.253384 8509 13@kv/kvserver/replicate_queue.go:842  [T1,n1,replicate,s1,r4/1:/System{/tsd-tse}] 719  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230402 14:00:00.266896 8072 3@pebble/event.go:717  [n1,s1,pebble] 720  [JOB 7] WAL created 000006
I230402 14:00:00.270623 7751 1@server/server_sql.go:1674  [T1,n1] 721  serving sql connections
I230402 14:00:00.271824 7751 testutils/testcluster/testcluster.go:1368  [-] 722  WaitForFullReplication
I230402 14:00:00.271864 7751 testutils/testcluster/testcluster.go:1372  [-] 723  WaitForFullReplication took: 298ns
E230402 14:00:00.284030 8327 sql/stats/automatic_stats.go:578  [T1,n1] 724  failed to get tables for automatic stats: injected error
    autocommit_extended_protocol_test.go:187: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/tests/autocommit_extended_protocol_test.go:187
        	Error:      	An error is expected but got nil.
        	Test:       	TestErrorDuringExtendedProtocolCommit
I230402 14:00:00.284204 7751 testutils/testcluster/testcluster.go:144  [-] 725  TestCluster quiescing nodes
I230402 14:00:00.284406 8327 sql/stats/automatic_stats.go:507  [T1,n1] 726  quiescing auto stats refresher
W230402 14:00:00.284401 8520 kv/kvserver/intentresolver/intent_resolver.go:826  [-] 727  failed to gc transaction record: could not GC completed transaction anchored at /Table/53/1/853192209781030913/"legacy_progress": node unavailable; try another peer
W230402 14:00:00.284449 8308 sql/sqlliveness/slinstance/slinstance.go:312  [n1] 728  exiting heartbeat loop
W230402 14:00:00.284659 8308 sql/sqlliveness/slinstance/slinstance.go:297  [n1] 730  exiting heartbeat loop
E230402 14:00:00.284687 8308 server/server_sql.go:504  [n1] 731  failed to run update of instance with new session ID: node unavailable; try another peer
E230402 14:00:00.284819 9500 jobs/registry.go:952  [T1,n1] 732  error getting live session: node unavailable; try another peer
I230402 14:00:00.284900 8381 jobs/registry.go:1522  [T1,n1] 734  AUTO SPAN CONFIG RECONCILIATION job 853192209781030913: stepping through state succeeded
I230402 14:00:00.284440 7801 server/start_listen.go:103  [T1,n1] 729  server shutting down: instructing cmux to stop accepting
I230402 14:00:00.284824 8824 rpc/context.go:2302  [T1,n1,rnode=1,raddr=127.0.0.1:38585,class=default,rpc] 733  connection heartbeat loop ended with err: <nil>
--- FAIL: TestErrorDuringExtendedProtocolCommit (1.31s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82010 sql/tests: TestErrorDuringExtendedProtocolCommit failed [C-test-failure O-robot T-sql-queries T-sql-sessions branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestErrorDuringExtendedProtocolCommit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestErrorDuringExtendedProtocolCommit [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9487762?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9487762?buildTab=artifacts#/) on master @ [8e6f530644eb35313a479af529b1f672c12dca38](https://github.com/cockroachdb/cockroach/commits/8e6f530644eb35313a479af529b1f672c12dca38):


```
I230407 13:48:29.703220 5671 testutils/testcluster/testcluster.go:1379  [-] 1029  WaitForFullReplication took: 500ns
I230407 13:48:29.710263 10380 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:77  [T1,n1,rangefeed=tenant-capability-watcher] 1030  Establishing MuxRangeFeed (/Table/8/{1-2}@1680875309.657811653,0...; 1 spans)
I230407 13:48:29.710417 10382 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:212  [T1,n1,rangefeed=tenant-capability-watcher] 1031  MuxRangeFeed starting for range /Table/8/{1-2}@1680875309.657811653,0 (rangeID 12, attempt 0)
I230407 13:48:29.710553 10383 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:297  [T1,n1,rangefeed=tenant-capability-watcher,mux_n=1,gen=2] 1032  Establishing MuxRangeFeed to node 1
I230407 13:48:29.712771 11058 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:77  [T1,n1,rangefeed=tenant-settings-watcher] 1033  Establishing MuxRangeFeed (/Table/5{0-1}@1680875309.657264576,0...; 1 spans)
I230407 13:48:29.713064 11060 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:212  [T1,n1,rangefeed=tenant-settings-watcher] 1034  MuxRangeFeed starting for range /Table/5{0-1}@1680875309.657264576,0 (rangeID 51, attempt 0)
I230407 13:48:29.713261 11061 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:297  [T1,n1,rangefeed=tenant-settings-watcher,mux_n=1,gen=2] 1035  Establishing MuxRangeFeed to node 1
E230407 13:48:29.716654 9313 sql/stats/automatic_stats.go:629  [T1,n1] 1036  failed to get tables for automatic stats: injected error
    autocommit_extended_protocol_test.go:187: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/tests/autocommit_extended_protocol_test.go:187
        	Error:      	An error is expected but got nil.
        	Test:       	TestErrorDuringExtendedProtocolCommit
I230407 13:48:29.717069 5671 testutils/testcluster/testcluster.go:145  [-] 1037  TestCluster quiescing nodes
I230407 13:48:29.717837 8838 server/start_listen.go:103  [T1,n1] 1039  server shutting down: instructing cmux to stop accepting
I230407 13:48:29.718145 9694 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,rangefeed=spanconfig-subscriber,mux_n=1,gen=2] 1040  MuxRangeFeed to node 1 terminating after 1.057254001s with err=recv msg error: recv: context canceled
I230407 13:48:29.718392 9691 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,rangefeed=spanconfig-subscriber] 1041  MuxRangeFeed  terminating after 1.057765172s with err=recv msg error: recv: context canceled
I230407 13:48:29.718574 10997 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:471  [T1,n1,rangefeed=tenant-settings-watcher] 1043  RangeFeed /Table/5{0-1}@1680875309.657264576,0 disconnected with last checkpoint 3.82844346s ago: node unavailable; try another peer
I230407 13:48:29.718717 9730 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,rangefeed=sql_instances,mux_n=1,gen=2] 1044  MuxRangeFeed to node 1 terminating after 1.052416138s with err=recv msg error: recv: context canceled
I230407 13:48:29.718805 9631 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,rangefeed=sql_instances] 1045  MuxRangeFeed  terminating after 1.052808409s with err=recv msg error: recv: context canceled
I230407 13:48:29.718863 9675 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,rangefeed=settings-watcher,mux_n=1,gen=2] 1046  MuxRangeFeed to node 1 terminating after 1.0573685s with err=recv msg error: recv: context canceled
I230407 13:48:29.718951 9700 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,rangefeed=settings-watcher] 1047  MuxRangeFeed  terminating after 1.057594815s with err=recv msg error: recv: context canceled
W230407 13:48:29.719013 9889 kv/kvserver/intentresolver/intent_resolver.go:826  [-] 1048  failed to gc transaction record: could not GC completed transaction anchored at /Table/3/1/102/2/1: node unavailable; try another peer
I230407 13:48:29.718483 11061 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,rangefeed=tenant-settings-watcher,mux_n=1,gen=2] 1042  MuxRangeFeed to node 1 terminating after 5.112503ms with err=recv msg error: recv: context canceled
I230407 13:48:29.719314 11058 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,rangefeed=tenant-settings-watcher] 1049  MuxRangeFeed  terminating after 6.309337ms with err=limiter begin: context canceled
I230407 13:48:29.719385 9338 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,rangefeed=lease,mux_n=1,gen=2] 1050  MuxRangeFeed to node 1 terminating after 1.101331677s with err=recv msg error: recv: context canceled
I230407 13:48:29.719452 10944 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,rangefeed=tenant-boundaries-watcher,mux_n=1,gen=2] 1051  MuxRangeFeed to node 1 terminating after 61.411198ms with err=recv msg error: recv: context canceled
I230407 13:48:29.719507 9818 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,job=AUTO SPAN CONFIG RECONCILIATION id=854605524564475905,rangefeed=sql-watcher-protected-ts-records-rangefeed,mux_n=1,gen=2] 1053  MuxRangeFeed to node 1 terminating after 930.972701ms with err=recv msg error: recv: context canceled
I230407 13:48:29.719535 9751 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,rangefeed=system-config-cache,mux_n=1,gen=3] 1054  MuxRangeFeed to node 1 terminating after 1.051176694s with err=recv msg error: recv: context canceled
I230407 13:48:29.719606 10673 rpc/context.go:2302  [T1,n1,rnode=1,raddr=127.0.0.1:42745,class=default,rpc] 1056  connection heartbeat loop ended with err: <nil>
I230407 13:48:29.719559 9926 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,job=AUTO SPAN CONFIG RECONCILIATION id=854605524564475905,rangefeed=sql-watcher-descriptor-rangefeed,mux_n=1,gen=2] 1057  MuxRangeFeed to node 1 terminating after 931.058027ms with err=recv msg error: recv: context canceled
I230407 13:48:29.719609 9917 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,job=AUTO SPAN CONFIG RECONCILIATION id=854605524564475905,rangefeed=sql-watcher-protected-ts-records-rangefeed] 1058  MuxRangeFeed  terminating after 931.298524ms with err=recv msg error: recv: context canceled
I230407 13:48:29.719617 9681 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,rangefeed=system-config-cache] 1059  MuxRangeFeed  terminating after 1.051427696s with err=recv msg error: recv: context canceled
I230407 13:48:29.719680 9923 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,job=AUTO SPAN CONFIG RECONCILIATION id=854605524564475905,rangefeed=sql-watcher-descriptor-rangefeed] 1060  MuxRangeFeed  terminating after 931.43418ms with err=recv msg error: recv: context canceled
W230407 13:48:29.719757 9287 sql/sqlliveness/slinstance/slinstance.go:334  [T1,n1] 1061  exiting heartbeat loop
W230407 13:48:29.719778 9287 sql/sqlliveness/slinstance/slinstance.go:321  [T1,n1] 1062  exiting heartbeat loop with error: node unavailable; try another peer
E230407 13:48:29.719809 9287 server/server_sql.go:510  [T1,n1] 1063  failed to run update of instance with new session ID: node unavailable; try another peer
I230407 13:48:29.719484 9335 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,rangefeed=lease] 1052  MuxRangeFeed  terminating after 1.101637801s with err=recv msg error: recv: context canceled
I230407 13:48:29.719549 10940 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,rangefeed=tenant-boundaries-watcher] 1055  MuxRangeFeed  terminating after 62.074044ms with err=recv msg error: recv: context canceled
I230407 13:48:29.719809 9182 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,rangefeed=table-stats-cache,mux_n=1,gen=2] 1064  MuxRangeFeed to node 1 terminating after 1.101455606s with err=recv msg error: recv: context canceled
I230407 13:48:29.719793 9836 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,job=AUTO SPAN CONFIG RECONCILIATION id=854605524564475905,rangefeed=sql-watcher-zones-rangefeed,mux_n=1,gen=2] 1065  MuxRangeFeed to node 1 terminating after 931.335763ms with err=recv msg error: recv: context canceled
I230407 13:48:29.719862 10383 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,rangefeed=tenant-capability-watcher,mux_n=1,gen=2] 1066  MuxRangeFeed to node 1 terminating after 9.251966ms with err=recv msg error: recv: context canceled
I230407 13:48:29.719896 9815 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,job=AUTO SPAN CONFIG RECONCILIATION id=854605524564475905,rangefeed=sql-watcher-zones-rangefeed] 1067  MuxRangeFeed  terminating after 931.624181ms with err=recv msg error: recv: context canceled
I230407 13:48:29.719890 9333 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,rangefeed=table-stats-cache] 1068  MuxRangeFeed  terminating after 1.101747887s with err=recv msg error: recv: context canceled
I230407 13:48:29.719941 10380 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,rangefeed=tenant-capability-watcher] 1069  MuxRangeFeed  terminating after 9.563528ms with err=recv msg error: recv: context canceled
I230407 13:48:29.717302 9313 sql/stats/automatic_stats.go:550  [T1,n1] 1038  quiescing auto stats refresher
I230407 13:48:29.719976 9722 jobs/registry.go:1538  [T1,n1] 1070  AUTO SPAN CONFIG RECONCILIATION job 854605524564475905: stepping through state succeeded
E230407 13:48:29.720165 9722 jobs/registry.go:943  [T1,n1] 1071  error getting live session: node unavailable; try another peer
W230407 13:48:29.726492 10684 jobs/adopt.go:518  [T1,n1] 1072  could not clear job claim: clear-job-claim: failed to send RPC: sending to all replicas failed; last error: ba: Scan [/Table/15/1/102,/Table/15/1/103), [txn: 0930614e], [can-forward-ts] RPC error: node unavailable; try another peer
W230407 13:48:29.728903 5671 server/server_sql.go:1704  [T1,n1] 1073  server shutdown without a prior graceful drain
--- FAIL: TestErrorDuringExtendedProtocolCommit (1.31s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #100480 sql/tests: TestErrorDuringExtendedProtocolCommit failed [C-test-failure O-robot T-sql-sessions branch-release-23.1]
- #82010 sql/tests: TestErrorDuringExtendedProtocolCommit failed [C-test-failure O-robot T-sql-queries T-sql-sessions branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestErrorDuringExtendedProtocolCommit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestErrorDuringExtendedProtocolCommit [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9512831?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9512831?buildTab=artifacts#/) on master @ [85e41ca8d3d9edacf5ee3061a2591b159b9b0502](https://github.com/cockroachdb/cockroach/commits/85e41ca8d3d9edacf5ee3061a2591b159b9b0502):


```
I230409 14:49:03.219095 10702 server/autoconfig/auto_config.go:97  [T1,n1,job=AUTO CONFIG RUNNER id=102] 839  using provider with type acprovider.NoTaskProvider
I230409 14:49:03.219138 10702 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 840  waiting for environment activation...
I230409 14:49:03.219173 10702 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 841  waiting for environment activation...
I230409 14:49:03.239408 8669 jobs/registry.go:608  [T1,n1] 842  active version is 1000023.1-2
I230409 14:49:03.247662 8669 upgrade/upgrademanager/manager.go:742  [T1,n1] 843  running Upgrade to 1000022.2-94: "create statement_activity and transaction_activity tables"
I230409 14:49:03.247733 10775 jobs/adopt.go:261  [T1,n1] 844  job 855183665460445185: resuming execution
I230409 14:49:03.263293 10750 jobs/registry.go:1538  [T1,n1] 845  MIGRATION job 855183665460445185: stepping through state running
I230409 14:49:03.274406 10750 jobs/registry.go:1538  [T1,n1] 846  MIGRATION job 855183665460445185: stepping through state succeeded
I230409 14:49:03.287139 10811 2@rpc/context.go:2565  [T1,n1,rnode=1,raddr=127.0.0.1:33747,class=default,rpc] 847  connection is now ready
I230409 14:49:03.295573 8669 jobs/wait.go:145  [T1,n1] 848  waited for 1 [855183665460445185] queued jobs to complete 47.860328ms
I230409 14:49:03.295780 8669 upgrade/upgrademanager/manager.go:283  [T1,n1] 849  running permanent upgrade for version 1000022.2-98
I230409 14:49:03.313818 8669 jobs/registry.go:608  [T1,n1] 850  active version is 1000023.1-2
I230409 14:49:03.321625 8669 upgrade/upgrademanager/manager.go:742  [T1,n1] 851  running Upgrade to 1000022.2-98: "change TTL for SQL Stats system tables"
I230409 14:49:03.321690 10843 jobs/adopt.go:261  [T1,n1] 852  job 855183665711775745: resuming execution
I230409 14:49:03.335169 10737 jobs/registry.go:1538  [T1,n1] 853  MIGRATION job 855183665711775745: stepping through state running
I230409 14:49:03.347851 10902 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=855183665711775745,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 854 ={"Timestamp":1681051743343226409,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230409 14:49:03.352668 10915 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=855183665711775745,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 855 ={"Timestamp":1681051743348162112,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230409 14:49:03.356184 10947 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=855183665711775745,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 856 ={"Timestamp":1681051743353039517,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230409 14:49:03.359907 10963 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=855183665711775745,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 857 ={"Timestamp":1681051743356521582,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230409 14:49:03.362398 10737 jobs/registry.go:1538  [T1,n1] 858  MIGRATION job 855183665711775745: stepping through state succeeded
I230409 14:49:03.381411 8669 jobs/wait.go:145  [T1,n1] 859  waited for 1 [855183665711775745] queued jobs to complete 59.738372ms
I230409 14:49:03.381479 8669 server/server_sql.go:1618  [T1,n1] 860  done ensuring all necessary startup migrations have run
I230409 14:49:03.381742 10993 jobs/job_scheduler.go:407  [T1,n1] 861  waiting 2m0s before scheduled jobs daemon start
I230409 14:49:03.382096 11015 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 862  tenant-settings-watcher: established range feed cache
I230409 14:49:03.382648 11017 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 863  tenant-capability-watcher: established range feed cache
I230409 14:49:03.383077 11017 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 864  received results of a full table scan for tenant capabilities
I230409 14:49:03.385554 11013 server/auto_upgrade.go:78  [T1,n1] 865  no need to upgrade, cluster already at the newest version
I230409 14:49:03.387581 11011 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 866  warmed privileges for virtual tables in 5.776331ms
I230409 14:49:03.425893 8992 3@pebble/event.go:717  [n1,s1,pebble] 867  [JOB 7] WAL created 000006
I230409 14:49:03.429728 8669 1@server/server_sql.go:1736  [T1,n1] 868  serving sql connections
I230409 14:49:03.431234 8669 testutils/testcluster/testcluster.go:1375  [-] 869  WaitForFullReplication
I230409 14:49:03.431272 8669 testutils/testcluster/testcluster.go:1379  [-] 870  WaitForFullReplication took: 336ns
E230409 14:49:03.445520 9369 sql/stats/automatic_stats.go:629  [T1,n1] 871  failed to get tables for automatic stats: injected error
I230409 14:49:03.445645 8669 testutils/testcluster/testcluster.go:145  [-] 872  TestCluster quiescing nodes
    autocommit_extended_protocol_test.go:187: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/tests/autocommit_extended_protocol_test.go:187
        	Error:      	An error is expected but got nil.
        	Test:       	TestErrorDuringExtendedProtocolCommit
I230409 14:49:03.445764 9369 sql/stats/automatic_stats.go:550  [T1,n1] 873  quiescing auto stats refresher
W230409 14:49:03.445926 10356 kv/kvserver/intentresolver/intent_resolver.go:826  [-] 874  failed to gc transaction record: could not GC completed transaction anchored at /Table/3/1/102/2/1: node unavailable; try another peer
I230409 14:49:03.446169 8876 server/start_listen.go:103  [T1,n1] 875  server shutting down: instructing cmux to stop accepting
W230409 14:49:03.446249 9255 sql/sqlliveness/slinstance/slinstance.go:334  [T1,n1] 877  exiting heartbeat loop
W230409 14:49:03.446280 9255 sql/sqlliveness/slinstance/slinstance.go:321  [T1,n1] 878  exiting heartbeat loop with error: node unavailable; try another peer
I230409 14:49:03.446216 10811 rpc/context.go:2302  [T1,n1,rnode=1,raddr=127.0.0.1:33747,class=default,rpc] 876  connection heartbeat loop ended with err: <nil>
E230409 14:49:03.446319 9255 server/server_sql.go:508  [T1,n1] 879  failed to run update of instance with new session ID: node unavailable; try another peer
I230409 14:49:03.446319 9788 jobs/registry.go:1538  [T1,n1] 880  AUTO SPAN CONFIG RECONCILIATION job 855183662642397185: stepping through state succeeded
E230409 14:49:03.446445 9788 jobs/registry.go:943  [T1,n1] 881  error getting live session: node unavailable; try another peer
W230409 14:49:03.453077 10702 jobs/adopt.go:518  [T1,n1] 882  could not clear job claim: clear-job-claim: failed to send RPC: sending to all replicas failed; last error: ba: Scan [/Table/15/1/102,/Table/15/1/103), [txn: d2012dc8], [can-forward-ts] RPC error: node unavailable; try another peer
W230409 14:49:03.457633 8669 server/server_sql.go:1696  [T1,n1] 883  server shutdown without a prior graceful drain
--- FAIL: TestErrorDuringExtendedProtocolCommit (1.28s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #100480 sql/tests: TestErrorDuringExtendedProtocolCommit failed [C-test-failure O-robot T-sql-sessions branch-release-23.1]
- #82010 sql/tests: TestErrorDuringExtendedProtocolCommit failed [C-test-failure O-robot T-sql-queries T-sql-sessions branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestErrorDuringExtendedProtocolCommit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/tests/BUILD.bazel
+++ b/pkg/sql/tests/BUILD.bazel
@@ -119,6 +119,7 @@ go_test(
         "//pkg/util/syncutil",
         "//pkg/util/timeutil",
         "//pkg/util/tracing",
+        "//pkg/util/tracing/tracingpb",
         "//pkg/util/uuid",
         "@com_github_cockroachdb_cockroach_go_v2//crdb",
         "@com_github_cockroachdb_cockroach_go_v2//crdb/crdbpgx",

--- a/pkg/sql/tests/autocommit_extended_protocol_test.go
+++ b/pkg/sql/tests/autocommit_extended_protocol_test.go
@@ -26,6 +26,8 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
+	"github.com/cockroachdb/cockroach/pkg/util/tracing"
+	"github.com/cockroachdb/cockroach/pkg/util/tracing/tracingpb"
 	"github.com/jackc/pgx/v4"
 	"github.com/stretchr/testify/require"
 )
@@ -156,18 +158,26 @@ func TestErrorDuringExtendedProtocolCommit(t *testing.T) {
 	var db *gosql.DB
 
 	var shouldErrorOnAutoCommit syncutil.AtomicBool
+	var traceID tracingpb.TraceID
 	params, _ := CreateTestServerParams()
 	params.Knobs.SQLExecutor = &sql.ExecutorTestingKnobs{
 		DisableAutoCommitDuringExec: true,
 		BeforeExecute: func(ctx context.Context, stmt string, descriptors *descs.Collection) {
 			if strings.Contains(stmt, "SELECT 'cat'") {
 				shouldErrorOnAutoCommit.Set(true)
+				traceID = tracing.SpanFromContext(ctx).TraceID()
 			}
 		},
 		BeforeAutoCommit: func(ctx context.Context, stmt string) error {
 			if shouldErrorOnAutoCommit.Get() {
 				shouldErrorOnAutoCommit.Set(false)
-				return errors.New("injected error")
+				// Only inject the error if we're in the same trace as the one we
+				// saw when executing our test query. This is so we know that this
+				// autocommit corresponds to our test qyery rather than an internal
+				// query.
+				if traceID == tracing.SpanFromContext(ctx).TraceID() {
+					return errors.New("injected error")
+				}
 			}
 			return nil
 		},

